
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Blast &#8212; TTS Research Technology Guides</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=8f2a1f02" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/graphviz.css?v=4ae1632d" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/bugfix.css?v=e4cc360a" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/gallery.css?v=c2bd73dd" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/navbar.css?v=3ee3464b" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/switcher.css?v=7c802369" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../_static/documentation_options.js?v=8d563738"></script>
    <script src="../../_static/doctools.js?v=9bcbadda"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'bio/apps/blast';</script>
    <script>
        DOCUMENTATION_OPTIONS.theme_version = '0.16.0';
        DOCUMENTATION_OPTIONS.theme_switcher_json_url = 'https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/switcher/switcher.json';
        DOCUMENTATION_OPTIONS.theme_switcher_version_match = '1.0.0';
        DOCUMENTATION_OPTIONS.show_version_warning_banner = true;
        </script>
    <link rel="canonical" href="https://rtguides.it.tufts.edu/bio/apps/blast.html" />
    <link rel="icon" href="../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Bowtie2" href="bowtie2.html" />
    <link rel="prev" title="Biopython" href="biopython.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="20250312" />
    <meta name="docbuild:last-update" content="Mar 12, 2025"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder=""
         aria-label=""
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
  <aside class="bd-header-announcement d-print-none d-none" aria-label="Announcement" data-pst-announcement-url="https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/announcement/announcement.html"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class="col-lg-3 navbar-header-items__start">
    
      <div class="navbar-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/jumbo.png" class="logo__image only-light" alt=""/>
    <img src="../../_static/jumbo.png" class="logo__image only-dark pst-js-only" alt=""/>
  
  
    <p class="title logo__title">TTS Research Technology Guides</p>
  
</a></div>
    
  </div>
  
  <div class="col-lg-9 navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
        </div>
      
      
        <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item"><nav class="bd-docs-nav bd-links" aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">
     Bioinformatics 
  </p>
  <div class="bd-toc-item navbar-nav"><ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../tutorials/index.html">Bioinformatics Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workshops/index.html">Bioinformatics Workshops</a></li>
<li class="toctree-l1"><a class="reference internal" href="../databases/index.html">Bioinformatics Databases</a></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="index.html">Bioinformatics Applications</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="abcreg.html">Abcreg</a></li>
<li class="toctree-l2"><a class="reference internal" href="abyss.html">Abyss</a></li>
<li class="toctree-l2"><a class="reference internal" href="alphafold.html">Alphafold</a></li>
<li class="toctree-l2"><a class="reference internal" href="amplify.html">Amplify</a></li>
<li class="toctree-l2"><a class="reference internal" href="angsd.html">Angsd</a></li>
<li class="toctree-l2"><a class="reference internal" href="bakta.html">Bakta</a></li>
<li class="toctree-l2"><a class="reference internal" href="bbmap.html">Bbmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="bbtools.html">Bbtools</a></li>
<li class="toctree-l2"><a class="reference internal" href="bcftools.html">Bcftools</a></li>
<li class="toctree-l2"><a class="reference internal" href="beast2.html">Beast2</a></li>
<li class="toctree-l2"><a class="reference internal" href="bedops.html">Bedops</a></li>
<li class="toctree-l2"><a class="reference internal" href="bedtools.html">Bedtools</a></li>
<li class="toctree-l2"><a class="reference internal" href="biobakery-workflows.html">Biobakery_workflows</a></li>
<li class="toctree-l2"><a class="reference internal" href="biopython.html">Biopython</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Blast</a></li>
<li class="toctree-l2"><a class="reference internal" href="bowtie2.html">Bowtie2</a></li>
<li class="toctree-l2"><a class="reference internal" href="breseq.html">Breseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="busco.html">Busco</a></li>
<li class="toctree-l2"><a class="reference internal" href="cactus.html">Cactus</a></li>
<li class="toctree-l2"><a class="reference internal" href="canu.html">Canu</a></li>
<li class="toctree-l2"><a class="reference internal" href="cellprofiler.html">Cellprofiler</a></li>
<li class="toctree-l2"><a class="reference internal" href="cellprofiler-analyst.html">Cellprofiler-analyst</a></li>
<li class="toctree-l2"><a class="reference internal" href="cellranger.html">Cellranger</a></li>
<li class="toctree-l2"><a class="reference internal" href="cellranger-atac.html">Cellranger-atac</a></li>
<li class="toctree-l2"><a class="reference internal" href="cellrank.html">Cellrank</a></li>
<li class="toctree-l2"><a class="reference internal" href="cufflinks.html">Cufflinks</a></li>
<li class="toctree-l2"><a class="reference internal" href="cutadapt.html">Cutadapt</a></li>
<li class="toctree-l2"><a class="reference internal" href="diamond.html">Diamond</a></li>
<li class="toctree-l2"><a class="reference internal" href="dorado.html">Dorado</a></li>
<li class="toctree-l2"><a class="reference internal" href="dragon-ora.html">Dragon_ora</a></li>
<li class="toctree-l2"><a class="reference internal" href="exonerate.html">Exonerate</a></li>
<li class="toctree-l2"><a class="reference internal" href="fasta3.html">Fasta3</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastp.html">Fastp</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastqc.html">Fastqc</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastspar.html">Fastspar</a></li>
<li class="toctree-l2"><a class="reference internal" href="fasttree.html">Fasttree</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtlong.html">Filtlong</a></li>
<li class="toctree-l2"><a class="reference internal" href="flye.html">Flye</a></li>
<li class="toctree-l2"><a class="reference internal" href="fqtk.html">Fqtk</a></li>
<li class="toctree-l2"><a class="reference internal" href="gatk4.html">Gatk4</a></li>
<li class="toctree-l2"><a class="reference internal" href="genomad.html">geNomad</a></li>
<li class="toctree-l2"><a class="reference internal" href="geomx-ngs-pipeline.html">Geomx_ngs_pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="guppy.html">Guppy</a></li>
<li class="toctree-l2"><a class="reference internal" href="hap-py.html">Hap.py</a></li>
<li class="toctree-l2"><a class="reference internal" href="hisat2.html">Hisat2</a></li>
<li class="toctree-l2"><a class="reference internal" href="hmmer.html">Hmmer</a></li>
<li class="toctree-l2"><a class="reference internal" href="homer.html">Homer</a></li>
<li class="toctree-l2"><a class="reference internal" href="htseq.html">Htseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="humann.html">Humann</a></li>
<li class="toctree-l2"><a class="reference internal" href="impute2.html">Impute2</a></li>
<li class="toctree-l2"><a class="reference internal" href="iqtree2.html">Iqtree2</a></li>
<li class="toctree-l2"><a class="reference internal" href="kallisto.html">Kallisto</a></li>
<li class="toctree-l2"><a class="reference internal" href="kneaddata.html">Kneaddata</a></li>
<li class="toctree-l2"><a class="reference internal" href="kraken2.html">Kraken2</a></li>
<li class="toctree-l2"><a class="reference internal" href="krakentools.html">Krakentools</a></li>
<li class="toctree-l2"><a class="reference internal" href="macs2.html">Macs2</a></li>
<li class="toctree-l2"><a class="reference internal" href="macs3.html">Macs3</a></li>
<li class="toctree-l2"><a class="reference internal" href="masurca.html">Masurca</a></li>
<li class="toctree-l2"><a class="reference internal" href="medaka.html">Medaka</a></li>
<li class="toctree-l2"><a class="reference internal" href="megahit.html">Megahit</a></li>
<li class="toctree-l2"><a class="reference internal" href="meme.html">Meme</a></li>
<li class="toctree-l2"><a class="reference internal" href="metaphlan.html">Metaphlan</a></li>
<li class="toctree-l2"><a class="reference internal" href="miniasm.html">Miniasm</a></li>
<li class="toctree-l2"><a class="reference internal" href="minimap2.html">Minimap2</a></li>
<li class="toctree-l2"><a class="reference internal" href="minipolish.html">Minipolish</a></li>
<li class="toctree-l2"><a class="reference internal" href="mirdeep2.html">Mirdeep2</a></li>
<li class="toctree-l2"><a class="reference internal" href="mirge3.html">Mirge3</a></li>
<li class="toctree-l2"><a class="reference internal" href="mothur.html">Mothur</a></li>
<li class="toctree-l2"><a class="reference internal" href="multiqc.html">Multiqc</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-ampliseq.html">Nf-core-ampliseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-atacseq.html">Nf-core-atacseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-bacass.html">Nf-core-bacass</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-bamtofastq.html">Nf-core-bamtofastq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-chipseq.html">Nf-core-chipseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-denovotranscript.html">Nf-core-denovotranscript</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-detaxizer.html">Nf-core-detaxizer</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-differentialabundance.html">Nf-core-differentialabundance</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-eager.html">Nf-core-eager</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-fetchngs.html">Nf-core-fetchngs</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-funcscan.html">Nf-core-funcscan</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-hic.html">Nf-core-hic</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-mag.html">Nf-core-mag</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-metatdenovo.html">Nf-core-metatdenovo</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-methylseq.html">Nf-core-methylseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-nanoseq.html">Nf-core-nanoseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-nanostring.html">Nf-core-nanostring</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-pangenome.html">Nf-core-pangenome</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-proteinfold.html">Nf-core-proteinfold</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-raredisease.html">Nf-core-raredisease</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-rnafusion.html">Nf-core-rnafusion</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-rnaseq.html">Nf-core-rnaseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-rnasplice.html">Nf-core-rnasplice</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-sarek.html">Nf-core-sarek</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-scrnaseq.html">Nf-core-scrnaseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-smrnaseq.html">Nf-core-smrnaseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-taxprofiler.html">Nf-core-taxprofiler</a></li>
<li class="toctree-l2"><a class="reference internal" href="nf-core-viralrecon.html">Nf-core-viralrecon</a></li>
<li class="toctree-l2"><a class="reference internal" href="orthofinder.html">Orthofinder</a></li>
<li class="toctree-l2"><a class="reference internal" href="pandaseq.html">Pandaseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="parabricks.html">Parabricks</a></li>
<li class="toctree-l2"><a class="reference internal" href="pepper-deepvariant.html">Pepper_deepvariant</a></li>
<li class="toctree-l2"><a class="reference internal" href="petitefinder.html">Petitefinder</a></li>
<li class="toctree-l2"><a class="reference internal" href="picard.html">Picard</a></li>
<li class="toctree-l2"><a class="reference internal" href="plink.html">Plink</a></li>
<li class="toctree-l2"><a class="reference internal" href="plink2.html">Plink2</a></li>
<li class="toctree-l2"><a class="reference internal" href="polypolish.html">Polypolish</a></li>
<li class="toctree-l2"><a class="reference internal" href="prokka.html">Prokka</a></li>
<li class="toctree-l2"><a class="reference internal" href="qiime2.html">Qiime2</a></li>
<li class="toctree-l2"><a class="reference internal" href="qualimap.html">Qualimap</a></li>
<li class="toctree-l2"><a class="reference internal" href="r-bioinformatics.html">R-bioinformatics</a></li>
<li class="toctree-l2"><a class="reference internal" href="r-scrnaseq.html">R-scrnaseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="r-shinyngs.html">R-shinyngs</a></li>
<li class="toctree-l2"><a class="reference internal" href="raven-assembler.html">Raven-assembler</a></li>
<li class="toctree-l2"><a class="reference internal" href="raxml-ng-mpi.html">Raxml-ng-mpi</a></li>
<li class="toctree-l2"><a class="reference internal" href="relion.html">Relion</a></li>
<li class="toctree-l2"><a class="reference internal" href="rmats2sashimiplot.html">Rmats2sashimiplot</a></li>
<li class="toctree-l2"><a class="reference internal" href="rnaquast.html">Rnaquast</a></li>
<li class="toctree-l2"><a class="reference internal" href="rosettafold2.html">Rosettafold2</a></li>
<li class="toctree-l2"><a class="reference internal" href="rosettafold2na.html">Rosettafold2na</a></li>
<li class="toctree-l2"><a class="reference internal" href="salmon.html">Salmon</a></li>
<li class="toctree-l2"><a class="reference internal" href="samtools.html">Samtools</a></li>
<li class="toctree-l2"><a class="reference internal" href="scanpy.html">Scanpy</a></li>
<li class="toctree-l2"><a class="reference internal" href="scvelo.html">Scvelo</a></li>
<li class="toctree-l2"><a class="reference internal" href="signalp6.html">Signalp6</a></li>
<li class="toctree-l2"><a class="reference internal" href="spaceranger.html">Spaceranger</a></li>
<li class="toctree-l2"><a class="reference internal" href="spades.html">Spades</a></li>
<li class="toctree-l2"><a class="reference internal" href="squid.html">Squid</a></li>
<li class="toctree-l2"><a class="reference internal" href="star.html">Star</a></li>
<li class="toctree-l2"><a class="reference internal" href="subread.html">Subread</a></li>
<li class="toctree-l2"><a class="reference internal" href="tmhmm.html">Tmhmm</a></li>
<li class="toctree-l2"><a class="reference internal" href="transdecoder.html">Transdecoder</a></li>
<li class="toctree-l2"><a class="reference internal" href="trgt.html">Trgt</a></li>
<li class="toctree-l2"><a class="reference internal" href="trim-galore.html">Trim-galore</a></li>
<li class="toctree-l2"><a class="reference internal" href="trimmomatic.html">Trimmomatic</a></li>
<li class="toctree-l2"><a class="reference internal" href="trinity.html">Trinity</a></li>
<li class="toctree-l2"><a class="reference internal" href="trinotate.html">Trinotate</a></li>
<li class="toctree-l2"><a class="reference internal" href="trycycler.html">Trycycler</a></li>
<li class="toctree-l2"><a class="reference internal" href="vcftools.html">Vcftools</a></li>
<li class="toctree-l2"><a class="reference internal" href="viennarna.html">Viennarna</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../lectures/index.html">Bioinformatics Lectures and Trainings</a></li>
<li class="toctree-l1"><a class="reference internal" href="../resources/index.html">Additional Resources</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"> 

<nav aria-label="Breadcrumb">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a
        href="../../index.html"
        class="nav-link"
        aria-label="Home"
      >
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
     
    <li class="breadcrumb-item">
      <a href="../index.html" class="nav-link"
        >Bioinformatics</a
      >
    </li>
     
    <li class="breadcrumb-item">
      <a href="index.html" class="nav-link"
        >Bioinformatics Applications</a
      >
    </li>
     
    <li class="breadcrumb-item active" aria-current="page">
      Blast
    </li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="blast">
<h1>Blast<a class="headerlink" href="#blast" title="Link to this heading">#</a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Link to this heading">#</a></h2>
<p>BLAST (Basic Local Alignment Search Tool) finds regions of similarity
between biological sequences. The program compares nucleotide or protein
sequences to sequence databases and calculates the statistical
significance.</p>
</section>
<section id="databases">
<h2>Databases<a class="headerlink" href="#databases" title="Link to this heading">#</a></h2>
<p>Local copies of blast databases can be found in the directory
<cite>/cluster/tufts/biocontainers/datasets/blast/latest/</cite>.</p>
<p>In blast modulefiles, we defined the environment variable <cite>BLASTDB</cite> as
<cite>/cluster/tufts/biocontainers/datasets/blast/latest</cite>, so that you can
use the blast databases such as <cite>nr</cite> and <cite>nt</cite> without specifying the
full path.</p>
</section>
<section id="versions">
<h2>Versions<a class="headerlink" href="#versions" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p>2.15.0</p></li>
<li><p>2.16.0</p></li>
</ul>
</section>
<section id="commands">
<h2>Commands<a class="headerlink" href="#commands" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p>amino-acid-composition</p></li>
<li><p>between-two-genes</p></li>
<li><p>blastdb_aliastool</p></li>
<li><p>blastdbcheck</p></li>
<li><p>blastdbcmd</p></li>
<li><p>blast_formatter</p></li>
<li><p>blastn</p></li>
<li><p>blastp</p></li>
<li><p>blastx</p></li>
<li><p>cleanup-blastdb-volumes.py</p></li>
<li><p>deltablast</p></li>
<li><p>dustmasker</p></li>
<li><p>eaddress</p></li>
<li><p>eblast</p></li>
<li><p>get_species_taxids.sh</p></li>
<li><p>legacy_blast.pl</p></li>
<li><p>makeblastdb</p></li>
<li><p>makembindex</p></li>
<li><p>makeprofiledb</p></li>
<li><p>psiblast</p></li>
<li><p>rpsblast</p></li>
<li><p>rpstblastn</p></li>
<li><p>run-ncbi-converter</p></li>
<li><p>segmasker</p></li>
<li><p>tblastn</p></li>
<li><p>tblastx</p></li>
<li><p>update_blastdb.pl</p></li>
<li><p>windowmasker</p></li>
</ul>
</section>
<section id="memory-requirements-for-blast">
<h2>Memory requirements for blast<a class="headerlink" href="#memory-requirements-for-blast" title="Link to this heading">#</a></h2>
<p>Ensure that you allocate sufficient memory for the BLAST database being
used. Insufficient memory will force the database to be repeatedly read
from disk, leading to intensive I/O operations that significantly slow
down your job. This can also cause severe strain on the filesystem,
especially when running multiple jobs concurrently. Such issues may
disrupt other users and prompt intervention from the Research Technology
(RT) staff.</p>
<p>Calculate the memory required by totalling the size of the .nsq (for
nucleotide databases) or .psq (for protein databases) files. e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ du -sh --total /fdb/blastdb/nt*.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.000.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.001.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.002.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.003.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.004.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.005.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.006.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.007.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.008.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.009.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.010.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.011.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.012.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.013.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.014.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.015.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.016.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.017.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.018.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.019.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.020.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.021.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.022.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.023.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.024.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.025.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.026.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.027.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.028.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.029.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.030.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.031.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.032.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.033.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.034.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.035.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.036.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.037.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.038.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.039.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.040.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.041.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.042.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.043.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.044.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.045.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.046.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.047.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.048.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.049.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.050.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.051.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.052.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.053.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.054.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.055.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.056.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.057.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.058.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.059.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.060.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.061.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.062.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.063.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.064.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.065.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.066.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.067.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.068.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.069.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.070.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.071.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.072.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.073.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.074.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.075.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.076.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.077.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.078.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.079.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.080.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.081.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.082.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.083.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.084.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.085.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.086.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.087.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.088.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.089.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.090.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.091.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.092.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.093.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.094.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.095.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.096.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.097.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.098.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.099.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.100.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.101.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.102.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.103.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.104.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.105.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.106.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.107.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.108.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.109.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.110.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.111.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.112.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.113.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.114.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.115.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.116.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.117.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.118.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.119.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.120.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.121.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.122.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.123.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.124.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.125.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.126.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.127.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.128.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.129.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.130.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.131.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.132.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.133.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.134.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.135.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.136.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.137.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.138.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.139.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.140.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.141.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.142.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.143.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.144.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.145.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.146.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.147.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.148.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.149.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.150.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.151.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.152.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.153.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.154.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.155.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.156.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.157.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.158.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.159.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.160.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.161.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.162.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.163.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.164.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.165.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.166.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.167.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.168.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.169.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.170.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.171.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.172.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.173.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.174.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.175.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.176.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.177.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.178.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.179.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.180.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.181.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.182.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.183.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.184.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.185.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.186.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.187.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.188.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.189.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.190.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.191.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.192.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.193.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.194.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.195.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.196.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.197.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.198.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.199.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.200.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.201.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.202.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.203.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.204.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.205.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.206.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.207.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.208.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.209.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.210.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.211.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.212.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.213.nsq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nt.214.nsq
760M        /cluster/tufts/biocontainers/datasets/blast/latest/nt.215.nsq
601G        total

$ du -sh --total /cluster/tufts/biocontainers/datasets/blast/latest/nr*.psq
3.4G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.000.psq
3.5G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.001.psq
3.2G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.002.psq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.003.psq
3.2G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.004.psq
3.1G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.005.psq
3.0G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.006.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.007.psq
47M     /cluster/tufts/biocontainers/datasets/blast/latest/nr.008.psq
3.2G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.009.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.010.psq
649M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.011.psq
3.2G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.012.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.013.psq
556M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.014.psq
3.3G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.015.psq
3.4G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.016.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.017.psq
756M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.018.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.019.psq
126M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.020.psq
3.6G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.021.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.022.psq
517M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.023.psq
3.1G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.024.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.025.psq
320M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.026.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.027.psq
970M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.028.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.029.psq
563M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.030.psq
2.6G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.031.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.032.psq
308M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.033.psq
3.6G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.034.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.035.psq
484M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.036.psq
3.5G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.037.psq
3.0G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.038.psq
3.0G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.039.psq
3.0G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.040.psq
3.3G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.041.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.042.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.043.psq
103M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.044.psq
2.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.045.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.046.psq
120M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.047.psq
3.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.048.psq
2.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.049.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.050.psq
62M     /cluster/tufts/biocontainers/datasets/blast/latest/nr.051.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.052.psq
113M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.053.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.054.psq
212M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.055.psq
2.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.056.psq
3.4G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.057.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.058.psq
384M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.059.psq
3.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.060.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.061.psq
96M     /cluster/tufts/biocontainers/datasets/blast/latest/nr.062.psq
3.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.063.psq
3.5G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.064.psq
3.6G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.065.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.066.psq
140M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.067.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.068.psq
140M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.069.psq
3.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.070.psq
3.1G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.071.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.072.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.073.psq
51M     /cluster/tufts/biocontainers/datasets/blast/latest/nr.074.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.075.psq
740M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.076.psq
3.3G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.077.psq
3.2G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.078.psq
3.4G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.079.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.080.psq
385M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.081.psq
3.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.082.psq
3.4G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.083.psq
3.1G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.084.psq
3.3G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.085.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.086.psq
269M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.087.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.088.psq
13M     /cluster/tufts/biocontainers/datasets/blast/latest/nr.089.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.090.psq
106M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.091.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.092.psq
390M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.093.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.094.psq
513M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.095.psq
2.9G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.096.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.097.psq
895M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.098.psq
3.2G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.099.psq
3.3G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.100.psq
3.3G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.101.psq
2.5G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.102.psq
2.9G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.103.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.104.psq
136M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.105.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.106.psq
3.2G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.107.psq
3.5G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.108.psq
3.5G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.109.psq
3.6G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.110.psq
3.1G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.111.psq
3.7G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.112.psq
3.1G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.113.psq
3.8G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.114.psq
702M        /cluster/tufts/biocontainers/datasets/blast/latest/nr.115.psq
1.3G        /cluster/tufts/biocontainers/datasets/blast/latest/nr.116.psq
302G        total
</pre></div>
</div>
<p>As you can see, if you want to run <strong>blastn</strong> with the <strong>nt</strong> database,
you need to allocate at least <strong>601GB</strong> of memory, or better yet,
<strong>610GB</strong> for safety.</p>
<p>If you want to run <strong>blastp</strong> with the <strong>nr</strong> database, you need to
allocate at least <strong>302GB</strong> of memory, or better yet, <strong>310GB</strong> for
safety.</p>
</section>
<section id="example-job">
<h2>Example job<a class="headerlink" href="#example-job" title="Link to this heading">#</a></h2>
<p>If you want to run a blast job with the <strong>nr</strong> or <strong>nr</strong> database, it is
recommended to use the <strong>largemem</strong> partition.</p>
<p>Adjust slurm options based on job requirements (<a class="reference external" href="https://slurm.schedmd.com/pdfs/summary.pdf">slurm cheat sheet</a>):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH -p largemem  # batch, gpu, preempt, mpi or your group&#39;s own partition</span>
<span class="c1">#SBATCH -t 1:00:00  # Runtime limit (D-HH:MM:SS)</span>
<span class="c1">#SBATCH -N 1   # Number of nodes</span>
<span class="c1">#SBATCH -n 1   # Number of tasks per node</span>
<span class="c1">#SBATCH -c 24  # Number of CPU cores per task</span>
<span class="c1">#SBATCH --mem=400G     # Memory required per node</span>
<span class="c1">#SBATCH --job-name=blast       # Job name</span>
<span class="c1">#SBATCH --mail-type=FAIL,BEGIN,END     # Send an email when job fails, begins, and finishes</span>
<span class="c1">#SBATCH --mail-user=your.email@tufts.edu       # Email address for notifications</span>
<span class="c1">#SBATCH --error=%x-%J-%u.err   # Standard error file: &lt;job_name&gt;-&lt;job_id&gt;-&lt;username&gt;.err</span>
<span class="c1">#SBATCH --output=%x-%J-%u.out  # Standard output file: &lt;job_name&gt;-&lt;job_id&gt;-&lt;username&gt;.out</span>

<span class="n">module</span> <span class="n">purge</span>   <span class="c1">### Optional, but highly recommended.</span>
<span class="n">module</span> <span class="n">load</span> <span class="n">blast</span><span class="o">/</span><span class="n">XXXX</span> <span class="c1">### Latest version is recommended.</span>

<span class="n">blastp</span> <span class="o">-</span><span class="n">query</span> <span class="n">query</span><span class="o">.</span><span class="n">fasta</span> <span class="o">-</span><span class="n">db</span> <span class="n">nr</span> <span class="o">-</span><span class="n">out</span> <span class="n">blastp_results</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">evalue</span> <span class="mf">1e-5</span> <span class="o">-</span><span class="n">outfmt</span> <span class="mi">6</span> <span class="o">-</span><span class="n">num_threads</span> <span class="mi">24</span>
</pre></div>
</div>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="biopython.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Biopython</p>
      </div>
    </a>
    <a class="right-next"
       href="bowtie2.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Bowtie2</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item"><div
  id="pst-page-navigation-heading-2"
  class="page-toc tocsection onthispage"
>
  <i class="fa-solid fa-list"></i> Blast
</div>
<nav
  class="bd-toc-nav page-toc"
  aria-labelledby="pst-page-navigation-heading-2"
>
  <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#introduction">Introduction</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#databases">Databases</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#versions">Versions</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#commands">Commands</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#memory-requirements-for-blast">Memory requirements for blast</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#example-job">Example job</a></li>
</ul>
</nav></div>

  <div class="sidebar-secondary-item">

  
  <div class="tocsection editthispage">
    <a href="https://github.com/TuftsRT/guides/edit/main/source/bio/apps/blast.rst">
      <i class="fa-solid fa-pencil"></i>
      
      
Suggest Edits 
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item">

  <div class="tocsection sourcelink">
    <a href="../../_sources/bio/apps/blast.rst.txt">
      <i class="fa-solid fa-file-lines"></i> Show Source
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item"><div class="tocsection report-error">
  <a
    href="https://github.com/TuftsRT/guides/issues/new?title=Error Report: bio/apps/blast"
    class="report-error-link"
    ><i class="fa-solid fa-triangle-exclamation"></i> Report Error
  </a>
</div></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">

  <p class="copyright">
    
       Copyright 2025, Tufts University.
      <br/>
    
  </p>
</div>
      
    </div>
  
  
    <div class="footer-items__center">
      
        <div class="footer-item"><p class="last-updated">
  Last updated on Mar 12, 2025.
  <br/>
</p></div>
      
    </div>
  
  
    <div class="footer-items__end">
      
        <div class="footer-item"><p class="switcher-label">Version:</p></div>
      
        <div class="footer-item">
<div class="version-switcher__container dropdown pst-js-only">
  <button id="pst-version-switcher-button-2"
    type="button"
    class="version-switcher__button btn btn-sm dropdown-toggle"
    data-bs-toggle="dropdown"
    aria-haspopup="listbox"
    aria-controls="pst-version-switcher-list-2"
    aria-label="Version switcher list"
  >
    Choose version  <!-- this text may get changed later by javascript -->
    <span class="caret"></span>
  </button>
  <div id="pst-version-switcher-list-2"
    class="version-switcher__menu dropdown-menu list-group-flush py-0"
    role="listbox" aria-labelledby="pst-version-switcher-button-2">
    <!-- dropdown will be populated by javascript on page load -->
  </div>
</div></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>