
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Introduction to Bioinformatics on Tufts HPC &#8212; TTS Research Technology Guides</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=8f2a1f02" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/graphviz.css?v=4ae1632d" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/bugfix.css?v=e4cc360a" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/gallery.css?v=c2bd73dd" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/navbar.css?v=3ee3464b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/css/switcher.css?v=7c802369" />
  
  <!-- So that users can add custom icons -->
  <script src="../../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../../_static/documentation_options.js?v=8313637b"></script>
    <script src="../../../_static/doctools.js?v=9bcbadda"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'bio/workshops/intro-to-bioinfo/02-bioinfo';</script>
    <script>
        DOCUMENTATION_OPTIONS.theme_version = '0.16.0';
        DOCUMENTATION_OPTIONS.theme_switcher_json_url = 'https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/switcher/switcher.json';
        DOCUMENTATION_OPTIONS.theme_switcher_version_match = '1.0.0-dev';
        DOCUMENTATION_OPTIONS.show_version_warning_banner = true;
        </script>
    <link rel="canonical" href="https://rtguides.it.tufts.edu/bio/workshops/intro-to-bioinfo/02-bioinfo.html" />
    <link rel="icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="Introduction to Singularity" href="../intro-to-singularity/index.html" />
    <link rel="prev" title="What is Linux?" href="01-linux.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="20250429" />
    <meta name="docbuild:last-update" content="Apr 29, 2025"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder=""
         aria-label=""
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
  <aside class="bd-header-announcement d-print-none d-none" aria-label="Announcement" data-pst-announcement-url="https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/announcement/announcement.html"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class="col-lg-3 navbar-header-items__start">
    
      <div class="navbar-item">

  
    
  

<a class="navbar-brand logo" href="../../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../../_static/jumbo.png" class="logo__image only-light" alt=""/>
    <img src="../../../_static/jumbo.png" class="logo__image only-dark pst-js-only" alt=""/>
  
  
    <p class="title logo__title">TTS Research Technology Guides</p>
  
</a></div>
    
  </div>
  
  <div class="col-lg-9 navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
        </div>
      
      
        <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item"><nav class="bd-docs-nav bd-links" aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">
     Bioinformatics 
  </p>
  <div class="bd-toc-item navbar-nav"><ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../tutorials/index.html">Bioinformatics Tutorials</a></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../index.html">Bioinformatics Workshops</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active has-children"><a class="reference internal" href="index.html">Beginner’s guide for conducting bioinformatics analysis on Tufts HPC</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="00-hpc.html">HPC Terminology</a></li>

<li class="toctree-l3"><a class="reference internal" href="01-linux.html">What is Linux?</a></li>
<li class="toctree-l3 current active"><a class="current reference internal" href="#">Introduction to Bioinformatics on Tufts HPC</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="../intro-to-singularity/index.html">Introduction to Singularity</a></li>
<li class="toctree-l2"><a class="reference internal" href="../intro-to-git-github.html">Introduction to Git &amp; GitHub</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../../databases/index.html">Bioinformatics Databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../apps/index.html">Bioinformatics Applications</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../lectures/index.html">Bioinformatics Lectures and Trainings</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../resources/index.html">Additional Resources</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"> 

<nav aria-label="Breadcrumb">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a
        href="../../../index.html"
        class="nav-link"
        aria-label="Home"
      >
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
     
    <li class="breadcrumb-item">
      <a href="../../index.html" class="nav-link"
        >Bioinformatics</a
      >
    </li>
     
    <li class="breadcrumb-item">
      <a href="../index.html" class="nav-link"
        >Bioinformatics Workshops</a
      >
    </li>
     
    <li class="breadcrumb-item">
      <a href="index.html" class="nav-link"
        >Beginner’s guide for...</a
      >
    </li>
     
    <li class="breadcrumb-item active" aria-current="page">
      Introduction to...
    </li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="introduction-to-bioinformatics-on-tufts-hpc">
<h1>Introduction to Bioinformatics on Tufts HPC<a class="headerlink" href="#introduction-to-bioinformatics-on-tufts-hpc" title="Link to this heading">#</a></h1>
<p>Author: Shirley Li, <a class="reference external" href="mailto:xue&#46;li37&#37;&#52;&#48;tufts&#46;edu">xue<span>&#46;</span>li37<span>&#64;</span>tufts<span>&#46;</span>edu</a></p>
<p>Date: 2025-02</p>
<section id="prerequisites">
<h2>Prerequisites<a class="headerlink" href="#prerequisites" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p>Basic understanding of biology and bioinformatics</p>
<ul>
<li><p>Familiarity with common bioinformatics data formats, including <a class="reference external" href="https://knowledge.illumina.com/software/general/software-general-reference_material-list/000002211">FASTQ</a>, <a class="reference external" href="https://pacbiofileformats.readthedocs.io/en/11.0/FASTA.html">FASTA</a>, <a class="reference external" href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM</a>, <a class="reference external" href="https://support.illumina.com/help/BS_App_RNASeq_Alignment_OLH_1000000006112/Content/Source/Informatics/BAM-Format.htm">BAM</a>, and <a class="reference external" href="https://useast.ensembl.org/info/website/upload/gff.html">GTF files</a>.</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://tuftsdatalab.github.io/tuftsWorkshops/2024_workshops/2024_bioinformatics201/linux/00_overview/">Familiarity with the command line</a></p></li>
<li><p><a class="reference external" href="https://it.tufts.edu/researchtechnology.tufts.edu">Access to an HPC cluster</a> (e.g., login credentials, necessary software installations)</p></li>
</ul>
</section>
<section id="bioinformatics-modules">
<h2>Bioinformatics modules<a class="headerlink" href="#bioinformatics-modules" title="Link to this heading">#</a></h2>
<section id="on-the-cluster">
<h3>On the cluster<a class="headerlink" href="#on-the-cluster" title="Link to this heading">#</a></h3>
<p>Use <code class="docutils literal notranslate"><span class="pre">module</span> <span class="pre">avail</span></code> to check the full list of tools available on the cluster. Below are some commonly used tools:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>   <span class="n">abcreg</span><span class="o">/</span><span class="mf">0.1.0</span>                         <span class="n">kallisto</span><span class="o">/</span><span class="mf">0.48.0</span>                     <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">orthofinder</span><span class="o">/</span><span class="mf">2.5.5</span>
   <span class="n">abyss</span><span class="o">/</span><span class="mf">2.3.7</span>                          <span class="n">kneaddata</span><span class="o">/</span><span class="mf">0.12.0</span>                           <span class="n">pandaseq</span><span class="o">/</span><span class="mf">2.11</span>
   <span class="n">alphafold</span><span class="o">/</span><span class="mf">2.3.0</span>                      <span class="n">kraken2</span><span class="o">/</span><span class="mf">2.1.3</span>                              <span class="n">parabricks</span><span class="o">/</span><span class="mf">4.0.0</span><span class="o">-</span><span class="mi">1</span>
   <span class="n">alphafold</span><span class="o">/</span><span class="mf">2.3.1</span>                      <span class="n">krakentools</span><span class="o">/</span><span class="mf">1.2</span>                            <span class="n">parabricks</span><span class="o">/</span><span class="mf">4.2.1</span><span class="o">-</span><span class="mi">1</span>
   <span class="n">alphafold</span><span class="o">/</span><span class="mf">2.3.2</span>                      <span class="n">macs2</span><span class="o">/</span><span class="mf">2.2.7.1</span>
   <span class="n">amplify</span><span class="o">/</span><span class="mf">2.0.0</span>                        <span class="n">macs3</span><span class="o">/</span><span class="mf">3.0.0</span><span class="n">a6</span>                              <span class="n">pepper_deepvariant</span><span class="o">/</span><span class="n">r0</span><span class="mf">.8</span>
   <span class="n">angsd</span><span class="o">/</span><span class="mf">0.939</span>                          <span class="n">masurca</span><span class="o">/</span><span class="mf">4.0.9</span>                              <span class="n">petitefinder</span><span class="o">/</span><span class="n">cpu</span>
   <span class="n">angsd</span><span class="o">/</span><span class="mf">0.940</span>                   <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">masurca</span><span class="o">/</span><span class="mf">4.1.0</span>                       <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">picard</span><span class="o">/</span><span class="mf">2.25.1</span>
   <span class="n">bakta</span><span class="o">/</span><span class="mf">1.9.3</span>                          <span class="n">medaka</span><span class="o">/</span><span class="mf">1.11.1</span>                              <span class="n">picard</span><span class="o">/</span><span class="mf">2.26.10</span>
   <span class="n">bbmap</span><span class="o">/</span><span class="mf">38.93</span>                          <span class="n">megahit</span><span class="o">/</span><span class="mf">1.2.9</span>                              <span class="n">plink</span><span class="o">/</span><span class="mf">1.90</span><span class="n">b6</span><span class="mf">.21</span>
   <span class="n">bbmap</span><span class="o">/</span><span class="mf">38.96</span>                   <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">meme</span><span class="o">/</span><span class="mf">5.5.5</span>                                 <span class="n">plink2</span><span class="o">/</span><span class="mf">2.00</span><span class="n">a2</span><span class="mf">.3</span>
   <span class="n">bbtools</span><span class="o">/</span><span class="mf">39.00</span>                        <span class="n">metaphlan</span><span class="o">/</span><span class="mf">4.0.2</span>                            <span class="n">polypolish</span><span class="o">/</span><span class="mf">0.5.0</span>
   <span class="n">bcftools</span><span class="o">/</span><span class="mf">1.13</span>                        <span class="n">metaphlan</span><span class="o">/</span><span class="mf">4.0.6</span>                     <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">preseq</span><span class="o">/</span><span class="mf">3.2.0</span>
   <span class="n">bcftools</span><span class="o">/</span><span class="mf">1.14</span>                        <span class="n">miniasm</span><span class="o">/</span><span class="mf">0.3</span><span class="n">_r179</span>                           <span class="n">prokka</span><span class="o">/</span><span class="mf">1.14.6</span>
   <span class="n">bcftools</span><span class="o">/</span><span class="mf">1.17</span>                        <span class="n">minimap2</span><span class="o">/</span><span class="mf">2.26</span>                       <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">qiime2</span><span class="o">/</span><span class="mf">2023.2</span>
   <span class="n">bcftools</span><span class="o">/</span><span class="mf">1.20</span>                 <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">minipolish</span><span class="o">/</span><span class="mf">0.1.3</span>                           <span class="n">qiime2</span><span class="o">/</span><span class="mf">2023.5</span>
   <span class="n">beast2</span><span class="o">/</span><span class="mf">2.6.3</span>                         <span class="n">mirdeep2</span><span class="o">/</span><span class="mf">2.0.1.3</span>                           <span class="n">qiime2</span><span class="o">/</span><span class="mf">2023.7</span>
   <span class="n">beast2</span><span class="o">/</span><span class="mf">2.6.4</span>                         <span class="n">mirge3</span><span class="o">/</span><span class="mf">0.1.4</span>                               <span class="n">qiime2</span><span class="o">/</span><span class="mf">2023.9</span>
   <span class="n">beast2</span><span class="o">/</span><span class="mf">2.6.6</span>                  <span class="p">(</span><span class="n">D</span><span class="p">)</span>    <span class="n">mothur</span><span class="o">/</span><span class="mf">1.46.0</span>                              <span class="n">qiime2</span><span class="o">/</span><span class="mf">2024.2</span>
   <span class="o">...</span> <span class="o">...</span>

</pre></div>
</div>
</section>
<section id="a-few-tips">
<h3>A few tips<a class="headerlink" href="#a-few-tips" title="Link to this heading">#</a></h3>
<ol class="arabic">
<li><p>Before installing your own tools, check if they are already available on the cluster using the <code class="docutils literal notranslate"><span class="pre">module</span> <span class="pre">avail</span></code> command.</p></li>
<li><p>Always be aware of the software versions, especially when using scripts from colleagues.</p></li>
<li><p>For less common tools, consider installing them yourself to ensure you have full control over the version and availability.</p>
<p><strong>If you need to install a less commonly used tool, it’s best to handle the installation yourself to ensure proper maintenance. Follow <a class="reference external" href="https://tuftsdatalab.github.io/tuftsWorkshops/2024_workshops/2024_bioinformatics301/01_source/">this tutorial</a> to install your own tool</strong></p>
</li>
</ol>
</section>
</section>
<section id="using-the-open-ondemand-app">
<h2>Using the Open OnDemand App<a class="headerlink" href="#using-the-open-ondemand-app" title="Link to this heading">#</a></h2>
<p>You can access Open OnDemand through <a class="reference external" href="https://ondemand.pax.tufts.edu/">this link</a></p>
<section id="bioinformatics-apps">
<h3>Bioinformatics Apps<a class="headerlink" href="#bioinformatics-apps" title="Link to this heading">#</a></h3>
<p>We offer a wide range of bioinformatics tools as apps, including <code class="docutils literal notranslate"><span class="pre">AlphaFold</span></code> and <code class="docutils literal notranslate"><span class="pre">CellProfiler</span></code>. Additionally, 31 nf-core pipelines are available as apps for ease of use, with the most popular being nf-core/rnaseq, which we will demonstrate in our final workshop.</p>
</section>
<section id="rstudio-and-apps">
<h3>RStudio and Apps<a class="headerlink" href="#rstudio-and-apps" title="Link to this heading">#</a></h3>
<p>RStudio Pax, use R/4.4.2 which has the most comprehensive packages installed (1300+).</p>
<p><strong>How to initiate an R job</strong></p>
<ol class="arabic simple">
<li><p>Log in to Open Ondemand.</p></li>
<li><p>Navigate to <code class="docutils literal notranslate"><span class="pre">interactive</span> <span class="pre">apps</span></code> and select <code class="docutils literal notranslate"><span class="pre">RStudio</span> <span class="pre">Pax</span></code></p></li>
<li><p>Specify the required resources:</p>
<ul class="simple">
<li><p><strong>Number of hours</strong></p></li>
<li><p><strong>Number of CPU cores</strong></p></li>
<li><p><strong>Amount of memory</strong></p></li>
<li><p><strong>CPU partition</strong> (set to <code class="docutils literal notranslate"><span class="pre">batch</span></code>)</p></li>
<li><p><strong>R version</strong> (latest available: 4.4.2)</p></li>
</ul>
</li>
<li><p>Click <code class="docutils literal notranslate"><span class="pre">Launch</span></code> to submit your job to the queue.</p></li>
<li><p>Wait a few minutes until your job starts running.</p></li>
<li><p>Click <code class="docutils literal notranslate"><span class="pre">Connect</span> <span class="pre">to</span> <span class="pre">Rstudio</span> <span class="pre">server</span></code></p></li>
<li><p>In RStudio, go to the <code class="docutils literal notranslate"><span class="pre">Packages</span></code> tab on the right to check the installed packages.</p></li>
</ol>
<p><strong>Installing R Packages</strong></p>
<p>Refer to our <a class="reference external" href="https://tuftsdatalab.github.io/tuftsWorkshops/2024_workshops/2024_bioinformatics301/02_Rpackage/">previous workshop materials</a> for detailed instructions on installing R packages.</p>
</section>
<section id="other-apps">
<h3>Other Apps<a class="headerlink" href="#other-apps" title="Link to this heading">#</a></h3>
<p>We also provide other applications like <code class="docutils literal notranslate"><span class="pre">Jupyter</span> <span class="pre">Bioinfo</span></code>, <code class="docutils literal notranslate"><span class="pre">JupyterLab</span></code>, <code class="docutils literal notranslate"><span class="pre">Jupyter</span> <span class="pre">Notebook</span></code>, <code class="docutils literal notranslate"><span class="pre">IGV</span></code>, and <code class="docutils literal notranslate"><span class="pre">Galaxy</span></code> to support your daily research activities.</p>
</section>
<section id="nf-core-pipelines">
<h3><a class="reference external" href="https://nf-co.re/pipelines">nf-core pipelines</a><a class="headerlink" href="#nf-core-pipelines" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>On cluster, use <code class="docutils literal notranslate"><span class="pre">module</span> <span class="pre">avail</span> <span class="pre">nf-core</span></code> to get the list of nf-core pipelines deployed on cluster.</p></li>
<li><p>On Open OnDemand, you can go to <code class="docutils literal notranslate"><span class="pre">bioinformatics</span> <span class="pre">apps</span></code> to find out what has been installed.</p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/tutorials/doc/Running_nfcore_pipelines_at_Tufts_HPC.html">Quick Start Guide to Using the nf-core Pipeline</a></p></li>
</ul>
</section>
</section>
<section id="writing-bioinformatics-job-script">
<h2>Writing Bioinformatics job script<a class="headerlink" href="#writing-bioinformatics-job-script" title="Link to this heading">#</a></h2>
<p>Let’s explore how to write a SLURM script, using STAR alignment as an example.</p>
<section id="prepare-the-slurm-script">
<h3>1. Prepare the SLURM Script<a class="headerlink" href="#prepare-the-slurm-script" title="Link to this heading">#</a></h3>
<p>Create a file named <code class="docutils literal notranslate"><span class="pre">run_star.sh</span></code> and add the following content:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>
<span class="c1">#SBATCH -J STAR_JOB             # Job name</span>
<span class="c1">#SBATCH --time=12:00:00         # Maximum runtime (D-HH:MM:SS format)</span>
<span class="c1">#SBATCH -p batch                # Partition (queue) to submit the job to</span>
<span class="c1">#SBATCH -n 1                    # Number of tasks (1 task in this case)</span>
<span class="c1">#SBATCH --mem=32g               # Memory allocation (32 GB)</span>
<span class="c1">#SBATCH --cpus-per-task=8       # Number of CPU cores allocated for the task</span>
<span class="c1">#SBATCH --output=STAR.%j.out    # Standard output file (%j = Job ID)</span>
<span class="c1">#SBATCH --error=STAR.%j.err     # Standard error file (%j = Job ID)</span>
<span class="c1">#SBATCH --mail-type=ALL         # Notifications for job status (start, end, fail)</span>
<span class="c1">#SBATCH --mail-user=utln@tufts.edu  # Your email address for notifications</span>

<span class="c1"># Load necessary module</span>
<span class="n">module</span> <span class="n">load</span> <span class="n">star</span><span class="o">/</span><span class="mf">2.7.11</span><span class="n">b</span>

<span class="c1"># Create output directory</span>
<span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="n">star_output</span>

<span class="c1"># Run STAR alignment for single-end reads</span>
<span class="n">STAR</span> <span class="o">--</span><span class="n">genomeDir</span> <span class="o">./</span><span class="n">reference_data</span><span class="o">/</span><span class="n">reference_index</span><span class="o">/</span> \
     <span class="o">--</span><span class="n">readFilesIn</span> <span class="o">./</span><span class="n">raw_fastq</span><span class="o">/</span><span class="n">Irrel_kd_1</span><span class="o">.</span><span class="n">subset</span><span class="o">.</span><span class="n">fq</span> \
     <span class="o">--</span><span class="n">outFileNamePrefix</span> <span class="o">./</span><span class="n">star_output</span><span class="o">/</span> \
     <span class="o">--</span><span class="n">runThreadN</span> <span class="mi">8</span>
</pre></div>
</div>
</section>
<section id="submit-the-job-script">
<h3>2. Submit the Job Script<a class="headerlink" href="#submit-the-job-script" title="Link to this heading">#</a></h3>
<p>If you’re running the script <strong>directly in the terminal</strong>, you need to make it executable first:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">chmod</span> <span class="o">+</span><span class="n">x</span> <span class="n">run_star</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>However, <strong>SLURM does not require execution permissions</strong>, so you can submit the job as-is using:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sbatch</span> <span class="n">run_star</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
</section>
<section id="monitor-job-status">
<h3>3. Monitor Job Status<a class="headerlink" href="#monitor-job-status" title="Link to this heading">#</a></h3>
<p>Use the following command to check the job status:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">squeue</span> <span class="o">-</span><span class="n">u</span> <span class="n">yourusername</span>
</pre></div>
</div>
</section>
<section id="star-workflow-details">
<h3>STAR Workflow Details<a class="headerlink" href="#star-workflow-details" title="Link to this heading">#</a></h3>
<section id="loading-modules">
<h4>Loading Modules<a class="headerlink" href="#loading-modules" title="Link to this heading">#</a></h4>
<p>Before using STAR, load the appropriate module:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">module</span> <span class="n">load</span> <span class="n">star</span><span class="o">/</span><span class="mf">2.7.11</span><span class="n">b</span>
</pre></div>
</div>
</section>
<section id="generating-the-genome-index">
<h4>Generating the Genome Index<a class="headerlink" href="#generating-the-genome-index" title="Link to this heading">#</a></h4>
<p>Before aligning reads, generate the genome index using a reference genome (<code class="docutils literal notranslate"><span class="pre">.fa</span></code>) and an annotation file (<code class="docutils literal notranslate"><span class="pre">.gtf</span></code>):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">STAR</span> <span class="o">--</span><span class="n">runMode</span> <span class="n">genomeGenerate</span> \
     <span class="o">--</span><span class="n">genomeDir</span> <span class="o">./</span><span class="n">reference_data</span><span class="o">/</span> \
     <span class="o">--</span><span class="n">genomeFastaFiles</span> <span class="o">./</span><span class="n">reference_data</span><span class="o">/</span><span class="n">chr1</span><span class="o">.</span><span class="n">fa</span> \
     <span class="o">--</span><span class="n">sjdbGTFfile</span> <span class="o">./</span><span class="n">reference_data</span><span class="o">/</span><span class="n">chr1</span><span class="o">-</span><span class="n">hg19_genes</span><span class="o">.</span><span class="n">gtf</span> \
     <span class="o">--</span><span class="n">runThreadN</span> <span class="mi">8</span>
</pre></div>
</div>
</section>
<section id="aligning-reads">
<h4>Aligning Reads<a class="headerlink" href="#aligning-reads" title="Link to this heading">#</a></h4>
<section id="for-single-end-reads">
<h5>For Single-End Reads:<a class="headerlink" href="#for-single-end-reads" title="Link to this heading">#</a></h5>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">STAR</span> <span class="o">--</span><span class="n">genomeDir</span> <span class="o">./</span><span class="n">reference_data</span><span class="o">/</span> \
     <span class="o">--</span><span class="n">readFilesIn</span> <span class="o">./</span><span class="n">raw_fastq</span><span class="o">/</span><span class="n">Irrel_kd_1</span><span class="o">.</span><span class="n">subset</span><span class="o">.</span><span class="n">fq</span> \
     <span class="o">--</span><span class="n">outFileNamePrefix</span> <span class="o">./</span><span class="n">star_output</span><span class="o">/</span> \
     <span class="o">--</span><span class="n">runThreadN</span> <span class="mi">8</span>
</pre></div>
</div>
</section>
<section id="for-paired-end-reads">
<h5>For Paired-End Reads:<a class="headerlink" href="#for-paired-end-reads" title="Link to this heading">#</a></h5>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">STAR</span> <span class="o">--</span><span class="n">genomeDir</span> <span class="o">./</span><span class="n">reference_data</span><span class="o">/</span> \
     <span class="o">--</span><span class="n">readFilesIn</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">read1</span><span class="o">.</span><span class="n">fastq</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">read2</span><span class="o">.</span><span class="n">fastq</span> \
     <span class="o">--</span><span class="n">outFileNamePrefix</span> <span class="o">./</span><span class="n">star_output</span><span class="o">/</span> \
     <span class="o">--</span><span class="n">runThreadN</span> <span class="mi">8</span>
</pre></div>
</div>
</section>
</section>
<section id="output-files">
<h4>Output Files<a class="headerlink" href="#output-files" title="Link to this heading">#</a></h4>
<p>After running STAR, the output directory will contain:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Aligned</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">sam</span>  <span class="n">Log</span><span class="o">.</span><span class="n">final</span><span class="o">.</span><span class="n">out</span>  <span class="n">Log</span><span class="o">.</span><span class="n">out</span>  <span class="n">Log</span><span class="o">.</span><span class="n">progress</span><span class="o">.</span><span class="n">out</span>  <span class="n">SJ</span><span class="o">.</span><span class="n">out</span><span class="o">.</span><span class="n">tab</span>
</pre></div>
</div>
<ul class="simple">
<li><p><strong><code class="docutils literal notranslate"><span class="pre">Aligned.out.sam</span></code></strong>: Contains alignment data.</p></li>
<li><p><strong><code class="docutils literal notranslate"><span class="pre">Log.final.out</span></code></strong>: Summarizes alignment metrics.</p></li>
</ul>
</section>
<section id="additional-tips">
<h4>Additional Tips<a class="headerlink" href="#additional-tips" title="Link to this heading">#</a></h4>
<ul class="simple">
<li><p>Always test commands interactively before incorporating them into job scripts.</p></li>
<li><p>Use the SLURM <code class="docutils literal notranslate"><span class="pre">--time</span></code>, <code class="docutils literal notranslate"><span class="pre">--mem</span></code>, and <code class="docutils literal notranslate"><span class="pre">--cpus-per-task</span></code> options to optimize resource allocation.</p></li>
<li><p>Check the SLURM output and error files for troubleshooting.</p></li>
</ul>
</section>
</section>
<section id="run-job-with-gpu-node">
<h3>Run job with GPU node<a class="headerlink" href="#run-job-with-gpu-node" title="Link to this heading">#</a></h3>
<section id="interactive-session">
<h4>Interactive session<a class="headerlink" href="#interactive-session" title="Link to this heading">#</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">srun</span> <span class="o">-</span><span class="n">p</span> <span class="n">preempt</span> <span class="o">-</span><span class="n">n</span> <span class="mi">1</span> <span class="o">--</span><span class="n">time</span><span class="o">=</span><span class="mi">04</span><span class="p">:</span><span class="mi">00</span><span class="p">:</span><span class="mi">00</span> <span class="o">--</span><span class="n">mem</span><span class="o">=</span><span class="mi">20</span><span class="n">G</span> <span class="o">--</span><span class="n">gres</span><span class="o">=</span><span class="n">gpu</span><span class="p">:</span><span class="mi">1</span> <span class="o">--</span><span class="n">pty</span> <span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">bash</span>
</pre></div>
</div>
<p>You can also specify which gpu node you would like to run jobs on</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">srun</span> <span class="o">-</span><span class="n">p</span> <span class="n">preempt</span> <span class="o">-</span><span class="n">n</span> <span class="mi">1</span> <span class="o">--</span><span class="n">time</span><span class="o">=</span><span class="mi">04</span><span class="p">:</span><span class="mi">00</span><span class="p">:</span><span class="mi">00</span> <span class="o">--</span><span class="n">mem</span><span class="o">=</span><span class="mi">20</span><span class="n">G</span> <span class="o">--</span><span class="n">gres</span><span class="o">=</span><span class="n">gpu</span><span class="p">:</span><span class="n">a100</span><span class="p">:</span><span class="mi">1</span> <span class="o">--</span><span class="n">pty</span> <span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">bash</span>
</pre></div>
</div>
</section>
</section>
<section id="submit-jobs-to-queue">
<h3>Submit jobs to queue<a class="headerlink" href="#submit-jobs-to-queue" title="Link to this heading">#</a></h3>
<p>Example script: <code class="docutils literal notranslate"><span class="pre">align.sh</span></code> using parabricks to do the alignment.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>#!/bin/bash
#SBATCH -J fq2bam_alignment          # Job name
#SBATCH -p preempt                   # Submit to the &#39;preempt&#39; partition (modify based on your cluster setup)
#SBATCH --gres=gpu:1                 # Request 1 GPU for accelerated processing
#SBATCH -n 2                         # Number of tasks (2 in this case)
#SBATCH --mem=60g                    # Memory allocation (60GB)
#SBATCH --time=02:00:00              # Maximum job run time (2 hours)
#SBATCH --cpus-per-task=20           # Number of CPU cores allocated per task
#SBATCH --output=alignment.%j.out    # Standard output file (with job ID %j)
#SBATCH --error=alignment.%j.err     # Standard error file (with job ID %j)
#SBATCH --mail-type=ALL              # Email notifications for all job states (begin, end, fail)
#SBATCH --mail-user=utln@tufts.edu   # Email address for notifications

# Load necessary modules
nvidia-smi                              # Show GPU information (optional for logging)
module load parabricks/4.0.0-1          # Load Parabricks module for GPU-accelerated alignment

# Define variables
genome_reference=&quot;/path/to/reference_genome&quot;      # Path to the reference genome (.fasta)
input_fastq1=&quot;/path/to/input_read1.fastq&quot;         # Path to the first paired-end FASTQ file
input_fastq2=&quot;/path/to/input_read2.fastq&quot;         # Path to the second paired-end FASTQ file
sample_name=&quot;sample_identifier&quot;                  # Sample identifier
known_sites_vcf=&quot;/path/to/known_sites.vcf&quot;        # Known sites VCF file for BQSR (optional, if available)
output_directory=&quot;/path/to/output_directory&quot;      # Directory for the output BAM file and reports
output_bam=&quot;${output_directory}/${sample_name}.bam&quot;            # Output BAM file path
output_bqsr_report=&quot;${output_directory}/${sample_name}.BQSR-report.txt&quot;  # Output BQSR report path

# Run the Parabricks fq2bam alignment pipeline
pbrun fq2bam \
    --ref ${genome_reference} \                # Reference genome (.fasta)
    --in-fq ${input_fastq1} ${input_fastq2} \  # Input paired-end FASTQ files
    --read-group-sm ${sample_name} \           # Sample name for read group
    --knownSites ${known_sites_vcf} \          # Known sites for BQSR
    --out-bam ${output_bam} \                  # Output BAM file
    --out-recal-file ${output_bqsr_report}     # Output Base Quality Score Recalibration (BQSR) report


</pre></div>
</div>
<p>Here is the command to submit job</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">chmod</span> <span class="o">+</span><span class="n">x</span> <span class="n">align</span><span class="o">.</span><span class="n">sh</span>    <span class="c1"># Makes the script executable</span>
<span class="n">sbatch</span> <span class="n">align</span><span class="o">.</span><span class="n">sh</span>      <span class="c1"># Submits the script to the SLURM queue</span>
</pre></div>
</div>
<p>Use <code class="docutils literal notranslate"><span class="pre">squeue</span> <span class="pre">-u</span> <span class="pre">yourusername</span></code> to check job status.</p>
</section>
</section>
<section id="additional-resources">
<h2>Additional Resources<a class="headerlink" href="#additional-resources" title="Link to this heading">#</a></h2>
<section id="datasets">
<h3>Datasets<a class="headerlink" href="#datasets" title="Link to this heading">#</a></h3>
<p>In bioinformatics, it’s common to download databases or reference genomes from public websites. For example, performing sequence alignment requires downloading the appropriate reference genome. To simplify this process, we have pre-downloaded and managed several databases/datasets for users. These include:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/alphafold.html">Alphafold2</a></p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/ncbi.html">NCBI BLAST</a></p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/diamond.html">Diamond</a></p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/igenomes.html">iGenomes</a></p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/kraken2.html">Kraken2</a></p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/biobakery.html">Biobakery</a></p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/genomad_db.html">geNomand</a></p></li>
<li><p><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/databases/doc/metaphlan.html">Metaphlan</a></p></li>
</ul>
</section>
<section id="new-user-guide">
<h3><a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/index.html">New user guide</a><a class="headerlink" href="#new-user-guide" title="Link to this heading">#</a></h3>
<p>In early 2025, we launched a new <a class="reference external" href="https://tuftsrt.github.io/guides/dev/index.html">RT Guides website</a>, offering comprehensive resources on a wide range of topics, including but not limited to HPC, data science, and and, <strong>most importantly, <a class="reference external" href="https://tuftsrt.github.io/guides/dev/bio/index.html">bioinformatics</a></strong>. We keep up with the latest trends and regularly update our materials to reflect new developments. We highly recommend bookmarking the website and referring to it whenever you encounter challenges. Your feedback is invaluable—let us know if you spot any errors or have suggestions.</p>
<p>For updates on bioinformatics education, software, and tools, consider <a class="reference external" href="https://elist.tufts.edu/sympa/info/best">subscribing</a> to our e-list: <a class="reference external" href="mailto:best&#37;&#52;&#48;elist&#46;tufts&#46;edu">best<span>&#64;</span>elist<span>&#46;</span>tufts<span>&#46;</span>edu</a>.</p>
</section>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="01-linux.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">What is Linux?</p>
      </div>
    </a>
    <a class="right-next"
       href="../intro-to-singularity/index.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Introduction to Singularity</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item"><div
  id="pst-page-navigation-heading-2"
  class="page-toc tocsection onthispage"
>
  <i class="fa-solid fa-list"></i> Introduction to Bioinformatics on Tufts HPC
</div>
<nav
  class="bd-toc-nav page-toc"
  aria-labelledby="pst-page-navigation-heading-2"
>
  <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#prerequisites">Prerequisites</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bioinformatics-modules">Bioinformatics modules</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#on-the-cluster">On the cluster</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#a-few-tips">A few tips</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#using-the-open-ondemand-app">Using the Open OnDemand App</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#bioinformatics-apps">Bioinformatics Apps</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#rstudio-and-apps">RStudio and Apps</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#other-apps">Other Apps</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#nf-core-pipelines">nf-core pipelines</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#writing-bioinformatics-job-script">Writing Bioinformatics job script</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-the-slurm-script">1. Prepare the SLURM Script</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#submit-the-job-script">2. Submit the Job Script</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#monitor-job-status">3. Monitor Job Status</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#star-workflow-details">STAR Workflow Details</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#loading-modules">Loading Modules</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#generating-the-genome-index">Generating the Genome Index</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#aligning-reads">Aligning Reads</a><ul class="nav section-nav flex-column">
<li class="toc-h5 nav-item toc-entry"><a class="reference internal nav-link" href="#for-single-end-reads">For Single-End Reads:</a></li>
<li class="toc-h5 nav-item toc-entry"><a class="reference internal nav-link" href="#for-paired-end-reads">For Paired-End Reads:</a></li>
</ul>
</li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#output-files">Output Files</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#additional-tips">Additional Tips</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#run-job-with-gpu-node">Run job with GPU node</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#interactive-session">Interactive session</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#submit-jobs-to-queue">Submit jobs to queue</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#additional-resources">Additional Resources</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#datasets">Datasets</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#new-user-guide">New user guide</a></li>
</ul>
</li>
</ul>
</nav></div>

  <div class="sidebar-secondary-item">

  
  <div class="tocsection editthispage">
    <a href="https://github.com/TuftsRT/guides/edit/develop/source/bio/workshops/intro-to-bioinfo/02-bioinfo.md">
      <i class="fa-solid fa-pencil"></i>
      
      
Suggest Edits 
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item">

  <div class="tocsection sourcelink">
    <a href="../../../_sources/bio/workshops/intro-to-bioinfo/02-bioinfo.md.txt">
      <i class="fa-solid fa-file-lines"></i> Show Source
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item"><div class="tocsection report-error">
  <a
    href="https://github.com/TuftsRT/guides/issues/new?title=Error Report: bio/workshops/intro-to-bioinfo/02-bioinfo"
    class="report-error-link"
    ><i class="fa-solid fa-triangle-exclamation"></i> Report Error
  </a>
</div></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">

  <p class="copyright">
    
      © Copyright 2025, Tufts University.
      <br/>
    
  </p>
</div>
      
    </div>
  
  
    <div class="footer-items__center">
      
        <div class="footer-item"><p class="last-updated">
  Last updated on Apr 29, 2025.
  <br/>
</p></div>
      
    </div>
  
  
    <div class="footer-items__end">
      
        <div class="footer-item"><p class="switcher-label">Version:</p></div>
      
        <div class="footer-item">
<div class="version-switcher__container dropdown pst-js-only">
  <button id="pst-version-switcher-button-2"
    type="button"
    class="version-switcher__button btn btn-sm dropdown-toggle"
    data-bs-toggle="dropdown"
    aria-haspopup="listbox"
    aria-controls="pst-version-switcher-list-2"
    aria-label="Version switcher list"
  >
    Choose version  <!-- this text may get changed later by javascript -->
    <span class="caret"></span>
  </button>
  <div id="pst-version-switcher-list-2"
    class="version-switcher__menu dropdown-menu list-group-flush py-0"
    role="listbox" aria-labelledby="pst-version-switcher-button-2">
    <!-- dropdown will be populated by javascript on page load -->
  </div>
</div></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>