---
tags: bioinformatics
---

# Bioinformatics Applications

Below are bioinformatics applications deployed by Tufts Research Technology.

```{gallery-grid}
---
grid-columns: 1
---
- header: "{fas}`dna` abcreg"
  content: "ABCreg is a tool used for automating approximate Bayesian computation by local linear regression."
  link: "abcreg.html"

- header: "{fas}`dna` abyss"
  content: "ABySS is a de novo sequence assembler intended for short paired-end reads and genomes of all sizes."
  link: "abyss.html"

- header: "{fas}`dna` alphafold"
  content: "Alphafold is an artificial intelligence program developed by Alphabets's/Google's DeepMind which performs predictions of protein structure."
  link: "alphafold.html"

- header: "{fas}`dna` amplify"
  content: "AMPlify is an attentive deep learning model for antimicrobial peptide prediction."
  link: "amplify.html"

- header: "{fas}`dna` angsd"
  content: "Angsd is a software for analyzing next generation sequencing data."
  link: "angsd.html"

- header: "{fas}`dna` bakta"
  content: "Bakta is a tool for the rapid & standardized annotation of bacterial genomes and plasmids from both isolates and MAGs. It provides dbxref-rich, sORF-including and taxon-independent annotations in machine-readable JSON & bioinformatics standard file formats for automated downstream analysis."
  link: "bakta.html"

- header: "{fas}`dna` bbmap"
  content: "Bbmap is a short read aligner, as well as various other bioinformatic tools."
  link: "bbmap.html"

- header: "{fas}`dna` bbtools"
  content: "BBTools is a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data."
  link: "bbtools.html"

- header: "{fas}`dna` bcftools"
  content: "Bcftools is a program for variant calling and manipulating files in the Variant Call Format (VCF) and its binary counterpart BCF."
  link: "bcftools.html"

- header: "{fas}`dna` beast2"
  content: "BEAST 2 is a cross-platform program for Bayesian phylogenetic analysis of molecular sequences."
  link: "beast2.html"

- header: "{fas}`dna` bedops"
  content: "Bedops is a software package for manipulating and analyzing genomic interval data."
  link: "bedops.html"

- header: "{fas}`dna` bedtools"
  content: "Bedtools is an extensive suite of utilities for genome arithmetic and comparing genomic features in BED format."
  link: "bedtools.html"

- header: "{fas}`dna` biobakery_workflows"
  content: "BioBakery workflows is a collection of workflows and tasks for executing common microbial community analyses using standardized, validated tools and parameters."
  link: "biobakery_workflows.html"

- header: "{fas}`dna` biopython"
  content: "Biopython is a set of freely available tools for biological computation written in Python."
  link: "biopython.html"

- header: "{fas}`dna` blast"
  content: "BLAST (Basic Local Alignment Search Tool) finds regions of similarity between biological sequences. The program compares nucleotide or protein sequences to sequence databases and calculates the statistical significance."
  link: "blast.html"

- header: "{fas}`dna` bowtie2"
  content: "Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s or 1,000s of characters, and particularly good at aligning to relatively long (e.g. mammalian) genomes. Bowtie 2 indexes the genome with an FM Index to keep its memory footprint small: for the human genome, its memory footprint is typically around 3.2 GB. Bowtie 2 supports gapped, local, and paired-end alignment modes."
  link: "bowtie2.html"

- header: "{fas}`dna` breseq"
  content: "Breseq is a computational pipeline for the analysis of short-read re-sequencing data."
  link: "breseq.html"

- header: "{fas}`dna` busco"
  content: "BUSCO (Benchmarking sets of Universal Single-Copy Orthologs) provides measures for quantitative assessment of genome assembly, gene set, and transcriptome completeness based on evolutionarily informed expectations of gene content from near-universal single-copy orthologs."
  link: "busco.html"

- header: "{fas}`dna` cactus"
  content: "Cactus is a reference-free whole-genome multiple alignment program."
  link: "cactus.html"

- header: "{fas}`dna` canu"
  content: "Canu is a fork of the Celera Assembler, designed for high-noise single-molecule sequencing (such as the PacBio RS II/Sequel or Oxford Nanopore MinION)."
  link: "canu.html"

- header: "{fas}`dna` cellprofiler"
  content: "CellProfiler is a free open-source software designed to enable biologists without training in computer vision or programming to quantitatively measure phenotypes from thousands of images automatically."
  link: "cellprofiler.html"

- header: "{fas}`dna` cellprofiler-analyst"
  content: "CellProfiler Analyst allows interactive exploration and analysis of data, particularly from high-throughput, image-based experiments."
  link: "cellprofiler-analyst.html"

- header: "{fas}`dna` cellranger"
  content: "Cellranger is a set of analysis pipelines that process Chromium single-cell data to align reads, generate feature-barcode matrices, perform clustering and other secondary analysis, and more."
  link: "cellranger.html"

- header: "{fas}`dna` cellranger-atac"
  content: "Cellranger-atac is a set of analysis pipelines that process Chromium Single Cell ATAC data."
  link: "cellranger-atac.html"

- header: "{fas}`dna` cellrank"
  content: "Cellrank is a toolkit to uncover cellular dynamics based on Markov state modeling of single-cell data."
  link: "cellrank.html"

- header: "{fas}`dna` cufflinks"
  content: "Cufflinks assembles transcripts, estimates their abundances, and tests for differential expression and regulation in RNA-Seq samples. It accepts aligned RNA-Seq reads and assembles the alignments into a parsimonious set of transcripts. Cufflinks then estimates the relative abundances of these transcripts based on how many reads support each one, taking into account biases in library preparation protocols."
  link: "cufflinks.html"

- header: "{fas}`dna` cutadapt"
  content: "Cutadapt finds and removes adapter sequences, primers, poly-A tails and other types of unwanted sequence from your high-throughput sequencing reads."
  link: "cutadapt.html"

- header: "{fas}`dna` diamond"
  content: "Diamond is sequence aligner for protein and translated DNA searches, designed for high performance analysis of big sequence data."
  link: "diamond.html"

- header: "{fas}`dna` dorado"
  content: "Dorado is a high-performance, easy-to-use, open source basecaller for Oxford Nanopore reads."
  link: "dorado.html"

- header: "{fas}`dna` dragon_ora"
  content: "The DRAGEN ORA Helper Suite Software is a suite of software for Linux distributions, designed to integrate in a transparent manner compressed FASTQ."
  link: "dragon_ora.html"

- header: "{fas}`dna` exonerate"
  content: "Exonerate is a generic tool for pairwise sequence comparison/alignment."
  link: "exonerate.html"

- header: "{fas}`dna` fasta3"
  content: "Fasta3 is a suite of programs for searching nucleotide or protein databases with a query sequence."
  link: "fasta3.html"

- header: "{fas}`dna` fastp"
  content: "Fastp is an ultra-fast all-in-one FASTQ preprocessor (QC/adapters/trimming/filtering/splitting/merging, etc)."
  link: "fastp.html"

- header: "{fas}`dna` fastqc"
  content: "FastQC aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines. It provides a modular set of analyses which you can use to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis."
  link: "fastqc.html"

- header: "{fas}`dna` fastspar"
  content: "FastSpar is a C++ implementation of the SparCC algorithm which is up to several thousand times faster than the original Python2 release and uses much less memory. The FastSpar implementation provides threading support and a p-value estimator which accounts for the possibility of repetitious data permutations."
  link: "fastspar.html"

- header: "{fas}`dna` fasttree"
  content: "FastTree infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences. FastTree can handle alignments with up to a million of sequences in a reasonable amount of time and memory. For large alignments, FastTree is 100-1,000 times faster than PhyML 3.0 or RAxML 7."
  link: "fasttree.html"

- header: "{fas}`dna` filtlong"
  content: "Filtlong is a tool for filtering long reads by quality. It can take a set of long reads and produce a smaller, better subset. It uses both read length (longer is better) and read identity (higher is better) when choosing which reads pass the filter."
  link: "filtlong.html"

- header: "{fas}`dna` flye"
  content: "Flye: Fast and accurate de novo assembler for single molecule sequencing reads"
  link: "flye.html"

- header: "{fas}`dna` fqtk"
  content: "fqtk is a toolkit for working with FASTQ files, written in Rust."
  link: "fqtk.html"

- header: "{fas}`dna` gatk4"
  content: "GATK (Genome Analysis Toolkit) is a collection of command-line tools for analyzing high-throughput sequencing data with a primary focus on variant discoverye."
  link: "gatk4.html"

- header: "{fas}`dna` genomad"
  content: "geNomad: Identification of mobile genetic elements."
  link: "genomad.html"

- header: "{fas}`dna` geomx_ngs_pipeline"
  content: "The GeoMx NGS Pipeline, developed by NanoString, is an essential part of the GeoMx NGS workflow."
  link: "geomx_ngs_pipeline.html"

- header: "{fas}`dna` guppy"
  content: "Guppy is a data processing toolkit that contains the Oxford Nanopore Technologies’ basecalling algorithms, and several bioinformatic post-processing features."
  link: "guppy.html"

- header: "{fas}`dna` hap.py"
  content: "Hap.py is a tool to compare diploid genotypes at haplotype level."
  link: "hap.py.html"

- header: "{fas}`dna` hisat2"
  content: "HISAT2 is a fast and sensitive alignment program for mapping next-generation sequencing reads (both DNA and RNA) to a population of human genomes as well as to a single reference genome."
  link: "hisat2.html"

- header: "{fas}`dna` hmmer"
  content: "Hmmer is used for searching sequence databases for sequence homologs, and for making sequence alignments."
  link: "hmmer.html"

- header: "{fas}`dna` homer"
  content: "HOMER is a suite of tools for Motif Discovery and next-gen sequencing analysis."
  link: "homer.html"

- header: "{fas}`dna` htseq"
  content: "HTSeq is a Python library to facilitate processing and analysis of data from high-throughput sequencing (HTS) experiments."
  link: "htseq.html"

- header: "{fas}`dna` humann"
  content: "Humann is a pipeline for efficiently and accurately profiling the presence/absence and abundance of microbial pathways in a community from metagenomic or metatranscriptomic sequencing data (typically millions of short DNA/RNA reads)."
  link: "humann.html"

- header: "{fas}`dna` impute2"
  content: "Impute2 is a genotype imputation and haplotype phasing program."
  link: "impute2.html"

- header: "{fas}`dna` iqtree2"
  content: "IQ-TREE is an efficient phylogenomic software by maximum likelihood."
  link: "iqtree2.html"

- header: "{fas}`dna` kallisto"
  content: "Kallisto is a program for quantifying abundances of transcripts from bulk and single-cell RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads."
  link: "kallisto.html"

- header: "{fas}`dna` kneaddata"
  content: "Kneaddata is a tool designed to perform quality control on metagenomic sequencing data."
  link: "kneaddata.html"

- header: "{fas}`dna` kraken2"
  content: "Kraken2 is a taxonomic sequence classifier that assigns taxonomic labels to DNA sequences."
  link: "kraken2.html"

- header: "{fas}`dna` krakentools"
  content: "Krakentools is a suite of scripts to be used for post-analysis of Kraken/KrakenUniq/Kraken2/Bracken results."
  link: "krakentools.html"

- header: "{fas}`dna` macs2"
  content: "MACS2 is Model-based Analysis of ChIP-Seq for identifying transcript factor binding sites."
  link: "macs2.html"

- header: "{fas}`dna` macs3"
  content: "Macs3 is Model-based Analysis of ChIP-Seq for identifying transcript factor."
  link: "macs3.html"

- header: "{fas}`dna` masurca"
  content: "The MaSuRCA (Maryland Super Read Cabog Assembler) genome assembly and analysis toolkit contains of MaSuRCA genome assembler, QuORUM error corrector for Illumina data, POLCA genome polishing software, Chromosome scaffolder, jellyfish mer counter, and MUMmer aligner."
  link: "masurca.html"

- header: "{fas}`dna` medaka"
  content: "Medaka is a tool to create consensus sequences and variant calls from nanopore sequencing data."
  link: "medaka.html"

- header: "{fas}`dna` megahit"
  content: "Megahit is a ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph."
  link: "megahit.html"

- header: "{fas}`dna` meme"
  content: "Meme is a collection of tools for the discovery and analysis of sequence motifs. Contents."
  link: "meme.html"

- header: "{fas}`dna` metaphlan"
  content: "Metaphlan is computational tool for profiling the composition of microbial communities (Bacteria, Archaea and Eukaryotes) from metagenomic shotgun sequencing data (i.e. not 16S) with species-level."
  link: "metaphlan.html"

- header: "{fas}`dna` miniasm"
  content: "Miniasm is a very fast OLC-based de novo assembler for noisy long reads."
  link: "miniasm.html"

- header: "{fas}`dna` minimap2"
  content: "Minimap2 is a versatile pairwise aligner for genomic and spliced nucleotide sequences."
  link: "minimap2.html"

- header: "{fas}`dna` minipolish"
  content: "Minipolish is a tool for Racon polishing of miniasm assemblies."
  link: "minipolish.html"

- header: "{fas}`dna` mirdeep2"
  content: "miRDeep2 discovers active known or novel miRNAs from deep sequencing data (Solexa/Illumina, 454, ...)."
  link: "mirdeep2.html"

- header: "{fas}`dna` mirge3"
  content: "Mirge3 is an update to Python package to perform comprehensive analysis of small RNA sequencing data, including miRNA annotation, A-to-I editing, novel miRNA detection, isomiR analysis, visualization through IGV, processing Unique Molecular Identifieres (UMI), tRF detection and producing interactive graphical output."
  link: "mirge3.html"

- header: "{fas}`dna` mothur"
  content: "Mothur is an open source software package for bioinformatics data processing."
  link: "mothur.html"

- header: "{fas}`dna` multiqc"
  content: "Multiqc is a reporting tool that parses summary statistics from results and log files generated by other bioinformatics tools."
  link: "multiqc.html"

- header: "{fas}`dna` nf-core-ampliseq"
  content: "nfcore/ampliseq is a bioinformatics analysis pipeline used for amplicon sequencing, supporting denoising of any amplicon and supports a variety of taxonomic databases for taxonomic assignment including 16S, ITS, CO1 and 18S. Phylogenetic placement is also possible. Multiple region analysis such as 5R is implemented. Supported is paired-end Illumina or single-end Illumina, PacBio and IonTorrent data. Default is the analysis of 16S rRNA gene amplicons sequenced paired-end with Illumina."
  link: "nf-core-ampliseq.html"

- header: "{fas}`dna` nf-core-atacseq"
  content: "nfcore/atacseq is a bioinformatics analysis pipeline used for ATAC-seq data."
  link: "nf-core-atacseq.html"

- header: "{fas}`dna` nf-core-bacass"
  content: "nf-core/bacass is a bioinformatics best-practice analysis pipeline for simple bacterial assembly and annotation. The pipeline is able to assemble short reads, long reads, or a mixture of short and long reads (hybrid assembly)."
  link: "nf-core-bacass.html"

- header: "{fas}`dna` nf-core-bamtofastq"
  content: "nf-core/bamtofastq is a bioinformatics best-practice analysis pipeline that converts (un)mapped .bam or .cram files into fq.gz files."
  link: "nf-core-bamtofastq.html"

- header: "{fas}`dna` nf-core-chipseq"
  content: "nfcore/chipseq is a bioinformatics analysis pipeline used for Chromatin ImmunopreciPitation sequencing (ChIP-seq) data."
  link: "nf-core-chipseq.html"

- header: "{fas}`dna` nf-core-denovotranscript"
  content: "nf-core/denovotranscript is a bioinformatics pipeline for de novo transcriptome assembly of paired-end short reads from bulk RNA-seq. It takes a samplesheet and FASTQ files as input, performs quality control (QC), trimming, assembly, redundancy reduction, pseudoalignment, and quantification. It outputs a transcriptome assembly FASTA file, a transcript abundance TSV file, and a MultiQC report with assembly quality and read QC metrics."
  link: "nf-core-denovotranscript.html"

- header: "{fas}`dna` nf-core-detaxizer"
  content: "nf-core/detaxizer is a pipeline to assess raw (meta)genomic data for contaminations and optionally filter reads which were classified as contamination. Default taxa classified as contamination are Homo and Homo sapiens."
  link: "nf-core-detaxizer.html"

- header: "{fas}`dna` nf-core-differentialabundance"
  content: "nf-core/differentialabundance is a bioinformatics pipeline that can be used to analyse data represented as matrices, comparing groups of observations to generate differential statistics and downstream analyses. The pipeline supports RNA-seq data such as that generated by the nf-core rnaseq workflow, and Affymetrix arrays via .CEL files."
  link: "nf-core-differentialabundance.html"

- header: "{fas}`dna` nf-core-eager"
  content: "nf-core/eager is a scalable and reproducible bioinformatics best-practise processing pipeline for genomic NGS sequencing data, with a focus on ancient DNA (aDNA) data. It is ideal for the (palaeo)genomic analysis of humans, animals, plants, microbes and even microbiomes."
  link: "nf-core-eager.html"

- header: "{fas}`dna` nf-core-fetchngs"
  content: "nf-core/fetchngs is a bioinformatics pipeline to fetch metadata and raw FastQ files from both public databases. At present, the pipeline supports SRA / ENA / DDBJ / GEO ids."
  link: "nf-core-fetchngs.html"

- header: "{fas}`dna` nf-core-funcscan"
  content: "nf-core/funcscan is a bioinformatics best-practice analysis pipeline for the screening of nucleotide sequences such as assembled contigs for functional genes. It currently features mining for antimicrobial peptides, antibiotic resistance genes and biosynthetic gene clusters."
  link: "nf-core-funcscan.html"

- header: "{fas}`dna` nf-core-hic"
  content: "nf-core/hic is a bioinformatics best-practice analysis pipeline for Analysis of Chromosome Conformation Capture data (Hi-C)."
  link: "nf-core-hic.html"

- header: "{fas}`dna` nf-core-mag"
  content: "nf-core/mag is a bioinformatics best-practise analysis pipeline for assembly, binning and annotation of metagenomes."
  link: "nf-core-mag.html"

- header: "{fas}`dna` nf-core-metatdenovo"
  content: "nf-core/metatdenovo is a bioinformatics best-practice analysis pipeline for assembly and annotation of metatranscriptomic data, both prokaryotic and eukaryotic.0"
  link: "nf-core-metatdenovo.html"

- header: "{fas}`dna` nf-core-methylseq"
  content: "nf-core/methylseq is a bioinformatics analysis pipeline used for Methylation (Bisulfite) sequencing data. It pre-processes raw data from FastQ inputs, aligns the reads and performs extensive quality-control on the results."
  link: "nf-core-methylseq.html"

- header: "{fas}`dna` nf-core-nanoseq"
  content: "nfcore/nanoseq is a bioinformatics analysis pipeline for Nanopore DNA/RNA sequencing data that can be used to perform basecalling, demultiplexing, QC, alignment, and downstream analysis."
  link: "nf-core-nanoseq.html"

- header: "{fas}`dna` nf-core-nanostring"
  content: "nf-core/nanostring is a bioinformatics pipeline that can be used to analyze NanoString data. The performed analysis steps include quality control and data normalization."
  link: "nf-core-nanostring.html"

- header: "{fas}`dna` nf-core-pangenome"
  content: "nf-core/pangenome is a bioinformatics best-practice analysis pipeline for pangenome graph construction. The pipeline renders a collection of sequences into a pangenome graph. Its goal is to build a graph that is locally directed and acyclic while preserving large-scale variation. Maintaining local linearity is important for interpretation, visualization, mapping, comparative genomics, and reuse of pangenome graphs."
  link: "nf-core-pangenome.html"

- header: "{fas}`dna` nf-core-proteinfold"
  content: "nf-core/proteinfold is a bioinformatics best-practice analysis pipeline for Protein 3D structure prediction."
  link: "nf-core-proteinfold.html"

- header: "{fas}`dna` nf-core-raredisease"
  content: "nf-core/raredisease is a best-practice bioinformatic pipeline for calling and scoring variants from WGS/WES data from rare disease patients."
  link: "nf-core-raredisease.html"

- header: "{fas}`dna` nf-core-rnafusion"
  content: "nf-core/rnafusion is a bioinformatics best-practice analysis pipeline for RNA sequencing consisting of several tools designed for detecting and visualizing fusion genes. Results from up to 5 fusion callers tools are created, and are also aggregated, most notably in a pdf visualiation document, a vcf data collection file, and html and tsv reports."
  link: "nf-core-rnafusion.html"

- header: "{fas}`dna` nf-core-rnaseq"
  content: "nf-core/rnaseq is a bioinformatics pipeline that can be used to analyse RNA sequencing data obtained from organisms with a reference genome and annotation. It takes a samplesheet and FASTQ files as input, performs quality control (QC), trimming and (pseudo-)alignment, and produces a gene expression matrix and extensive QC report."
  link: "nf-core-rnaseq.html"

- header: "{fas}`dna` nf-core-rnasplice"
  content: "nf-core/rnasplice is a bioinformatics pipeline for alternative splicing analysis of RNA sequencing data obtained from organisms with a reference genome and annotation."
  link: "nf-core-rnasplice.html"

- header: "{fas}`dna` nf-core-sarek"
  content: "nf-core/sarek is a workflow designed to detect variants on whole genome or targeted sequencing data. Initially designed for Human, and Mouse, it can work on any species with a reference genome. Sarek can also handle tumour / normal pairs and could include additional relapses."
  link: "nf-core-sarek.html"

- header: "{fas}`dna` nf-core-scrnaseq"
  content: "nf-core/scrnaseq is a bioinformatics best-practice analysis pipeline for processing 10x Genomics single-cell RNA-seq data."
  link: "nf-core-scrnaseq.html"

- header: "{fas}`dna` nf-core-smrnaseq"
  content: "nf-core/smrnaseq is a bioinformatics best-practice analysis pipeline for Small RNA-Seq."
  link: "nf-core-smrnaseq.html"

- header: "{fas}`dna` nf-core-taxprofiler"
  content: "nf-core/taxprofiler is a bioinformatics best-practice analysis pipeline for taxonomic classification and profiling of shotgun short- and long-read metagenomic data. It allows for in-parallel taxonomic identification of reads or taxonomic abundance estimation with multiple classification and profiling tools against multiple databases, and produces standardised output tables for facilitating results comparison between different tools and databases."
  link: "nf-core-taxprofiler.html"

- header: "{fas}`dna` nf-core-viralrecon"
  content: "nf-core/viralrecon is a bioinformatics analysis pipeline used to perform assembly and intra-host/low-frequency variant calling for viral samples. The pipeline supports both Illumina and Nanopore sequencing data."
  link: "nf-core-viralrecon.html"

- header: "{fas}`dna` orthofinder"
  content: "Orthofinder is a fast, accurate and comprehensive platform for comparative genomics. It finds orthogroups and orthologs, infers rooted gene trees for all orthogroups and identifies all of the gene duplication events in those gene trees. It also infers a rooted species tree for the species being analysed and maps the gene duplication events from the gene trees to branches in the species tree. OrthoFinder also provides comprehensive statistics for comparative genomic analyses."
  link: "orthofinder.html"

- header: "{fas}`dna` pandaseq"
  content: "Pandaseq is a program to align Illumina reads, optionally with PCR primers embedded in the sequence, and reconstruct an overlapping sequence."
  link: "pandaseq.html"

- header: "{fas}`dna` parabricks"
  content: "NVIDIA's Clara Parabricks brings next generation sequencing to GPUs, accelerating an array of gold-standard tooling such as BWA-MEM, GATK4, Google's DeepVariant, and many more. Users can achieve a 30-60x acceleration and 99.99% accuracy for variant calling when comparing against CPU-only BWA-GATK4 pipelines, meaning a single server can process up to 60 whole genomes per day. These tools can be easily integrated into current pipelines with drop-in replacement commands to quickly bring speed and data-center scale to a range of applications including germline, somatic and RNA workflows."
  link: "parabricks.html"

- header: "{fas}`dna` pepper_deepvariant"
  content: "PEPPER is a genome inference module based on recurrent neural networks that enables long-read variant calling and nanopore assembly polishing in the PEPPER-Margin-DeepVariant pipeline."
  link: "pepper_deepvariant.html"

- header: "{fas}`dna` petitefinder"
  content: "petiteFinder is an automated computer vision tool to compute Petite colony frequencies in baker's yeast."
  link: "petitefinder.html"

- header: "{fas}`dna` picard"
  content: "Picard is a set of command line tools for manipulating high-throughput sequencing (HTS) data and formats such as SAM/BAM/CRAM and VCF."
  link: "picard.html"

- header: "{fas}`dna` plink"
  content: "Plink is a free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner."
  link: "plink.html"

- header: "{fas}`dna` plink2"
  content: "Plink2 is a whole genome association analysis toolset."
  link: "plink2.html"

- header: "{fas}`dna` polypolish"
  content: "Polypolish is a tool for polishing genome assemblies with short reads."
  link: "polypolish.html"

- header: "{fas}`dna` prokka"
  content: "Prokka is a software tool to annotate bacterial, archaeal and viral genomes quickly and produce standards-compliant output files."
  link: "prokka.html"

- header: "{fas}`dna` qiime2"
  content: "QIIME 2 is a powerful, extensible, and decentralized microbiome analysis package with a focus on data and analysis transparency. QIIME 2 enables researchers to start an analysis with raw DNA sequence data and finish with publication-quality figures and statistical results."
  link: "qiime2.html"

- header: "{fas}`dna` qualimap"
  content: "Qualimap is a platform-independent application written in Java and R that provides both a Graphical User Interface (GUI) and a command-line interface to facilitate the quality control of alignment sequencing data and its derivatives like feature counts."
  link: "qualimap.html"

- header: "{fas}`dna` r-bioinformatics"
  content: "RStudio is an integrated development environment (IDE) for the R statistical computation and graphics system."
  link: "r-bioinformatics.html"

- header: "{fas}`dna` r-scrnaseq"
  content: "RStudio is an integrated development environment (IDE) for the R statistical computation and graphics system."
  link: "r-scrnaseq.html"

- header: "{fas}`dna` r-shinyngs"
  content: "Shinyngs is an R package designed to facilitate downstream analysis of RNA-seq and similar expression data with various exploratory plots and data mining tools."
  link: "r-shinyngs.html"

- header: "{fas}`dna` raven-assembler"
  content: "Raven-assembler is a de novo genome assembler for long uncorrected reads."
  link: "raven-assembler.html"

- header: "{fas}`dna` raxml-ng-mpi"
  content: "Raxml-ng is a phylogenetic tree inference tool which uses maximum-likelihood (ML) optimality criterion."
  link: "raxml-ng-mpi.html"

- header: "{fas}`dna` relion"
  content: "RELION (for REgularised LIkelihood OptimisatioN) is a stand-alone computer program for Maximum A Posteriori refinement of (multiple) 3D reconstructions or 2D class averages in cryo-electron microscopy. It is developed in the research group of Sjors Scheres at the MRC Laboratory of Molecular Biology."
  link: "relion.html"

- header: "{fas}`dna` rmats2sashimiplot"
  content: "Rmats2sashimiplot produces a sashimiplot visualization of rMATS output."
  link: "rmats2sashimiplot.html"

- header: "{fas}`dna` rnaquast"
  content: "Rnaquast is a quality assessment tool for de novo transcriptome assemblies."
  link: "rnaquast.html"

- header: "{fas}`dna` rosettafold2"
  content: "RoseTTAFold2 extends the original three-track architecture of RoseTTAFold over the full network, incorporating the concepts of Frame-aligned point error, recycling during training, and the use of a distillation set from AlphaFold2."
  link: "rosettafold2.html"

- header: "{fas}`dna` rosettafold2na"
  content: "RoseTTAFoldNA rapidly produces three-dimensional structure models with confidence estimates for protein–DNA and protein–RNA complexes."
  link: "rosettafold2na.html"

- header: "{fas}`dna` salmon"
  content: "Salmon is a tool for quantifying the expression of transcripts using RNA-seq data."
  link: "salmon.html"

- header: "{fas}`dna` samtools"
  content: "Samtools is a set of utilities for the Sequence Alignment/Map (SAM) format."
  link: "samtools.html"

- header: "{fas}`dna` scanpy"
  content: "Scanpy is a scalable toolkit for analyzing single-cell gene expression data built jointly with anndata."
  link: "scanpy.html"

- header: "{fas}`dna` scvelo"
  content: "Scvelo is a scalable toolkit for RNA velocity analysis in single cells."
  link: "scvelo.html"

- header: "{fas}`dna` signalp6"
  content: "SignalP predicts the presence and location of signal peptide cleavage sites in amino acid sequences from different organisms: Gram-positive prokaryotes, Gram-negative prokaryotes, and eukaryotes"
  link: "signalp6.html"

- header: "{fas}`dna` spaceranger"
  content: "Spaceranger is a set of analysis pipelines that process Visium Spatial Gene Expression data with brightfield and fluorescence microscope images."
  link: "spaceranger.html"

- header: "{fas}`dna` spades"
  content: "Spades is an assembly toolkit containing various assembly pipelines."
  link: "spades.html"

- header: "{fas}`dna` squid"
  content: "SQUID is designed to detect both fusion-gene and non-fusion-gene transcriptomic structural variations from RNA-seq alignment."
  link: "squid.html"

- header: "{fas}`dna` star"
  content: "STAR (Spliced Transcripts Alignment to a Reference) is an ultrafast universal RNA-seq aligner."
  link: "star.html"

- header: "{fas}`dna` subread"
  content: "Subread carries out high-performance read alignment, quantification and mutation discovery. It is a general-purpose read aligner which can be used to map both genomic DNA-seq reads and DNA-seq reads. It uses a new mapping paradigm called seed-and-vote to achieve fast, accurate and scalable read mapping. Subread automatically determines if a read should be globally or locally aligned, therefore particularly powerful in mapping RNA-seq reads. It supports INDEL detection and can map reads with both fixed and variable lengths."
  link: "subread.html"

- header: "{fas}`dna` tmhmm"
  content: "Tmhmm is used for prediction of transmembrane helices in proteins."
  link: "tmhmm.html"

- header: "{fas}`dna` transdecoder"
  content: "Transdecoder identifies candidate coding regions within transcript sequences, such as those generated by de novo RNA-Seq transcript assembly using Trinity, or constructed based on RNA-Seq alignments to the genome using Tophat and Cufflinks."
  link: "transdecoder.html"

- header: "{fas}`dna` trgt"
  content: "TRGT is a tool for targeted genotyping of tandem repeats from PacBio HiFi data. In addition to the basic size genotyping, TRGT profiles sequence composition, mosaicism, and CpG methylation of each analyzed repeat and visualization of reads overlapping the repeats."
  link: "trgt.html"

- header: "{fas}`dna` trim-galore"
  content: "Trim-galore is a wrapper tool that automates quality and adapter trimming to FastQ files."
  link: "trim-galore.html"

- header: "{fas}`dna` trimmomatic"
  content: "Trimmomatic is a flexible read trimming tool for Illumina NGS data"
  link: "trimmomatic.html"

- header: "{fas}`dna` trinity"
  content: "Trinity assembles transcript sequences from Illumina RNA-Seq data."
  link: "trinity.html"

- header: "{fas}`dna` trinotate"
  content: "Trinotate is a comprehensive annotation suite designed for automatic functional annotation of transcriptomes, particularly de novo assembled transcriptomes, from model or non-model organisms."
  link: "trinotate.html"

- header: "{fas}`dna` trycycler"
  content: "Trycycler is a tool for generating consensus long-read assemblies for bacterial genomes."
  link: "trycycler.html"

- header: "{fas}`dna` vcftools"
  content: "VCFtools is a program package designed for working with VCF files, such as those generated by the 1000 Genomes Project. The aim of VCFtools is to provide easily accessible methods for working with complex genetic variation data in the form of VCF files."
  link: "vcftools.html"

- header: "{fas}`dna` viennarna"
  content: "Viennarna is a set of standalone programs and libraries used for prediction and analysis of RNA secondary structures."
  link: "viennarna.html"

```
