
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Run AlphaFold2 on Tufts HPC with Open OnDemand App &#8212; TTS Research Technology Guides</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=8f2a1f02" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/graphviz.css?v=4ae1632d" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/style/bugfix.css?v=736ec69d" />
    <link rel="stylesheet" type="text/css" href="../../../_static/style/footer.css?v=5f497d55" />
    <link rel="stylesheet" type="text/css" href="../../../_static/style/gallery.css?v=c2bd73dd" />
    <link rel="stylesheet" type="text/css" href="../../../_static/style/navbar.css?v=147f7454" />
    <link rel="stylesheet" type="text/css" href="../../../_static/style/sidebar.css?v=ff9f960e" />
    <link rel="stylesheet" type="text/css" href="../../../_static/style/switcher.css?v=3e40d84b" />
  
  <!-- So that users can add custom icons -->
  <script src="../../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../../_static/documentation_options.js?v=8d563738"></script>
    <script src="../../../_static/doctools.js?v=9bcbadda"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-ZJ5LXFRJ2G"></script>
    <script>
                window.dataLayer = window.dataLayer || [];
                function gtag(){ dataLayer.push(arguments); }
                gtag('js', new Date());
                gtag('config', 'G-ZJ5LXFRJ2G');
            </script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'bio/lectures/introduction-to-alphafold2/02-run-alpha-fold2-open-ondemand';</script>
    <script>
        DOCUMENTATION_OPTIONS.theme_version = '0.16.0';
        DOCUMENTATION_OPTIONS.theme_switcher_json_url = 'https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/switcher/switcher.json';
        DOCUMENTATION_OPTIONS.theme_switcher_version_match = '1.0.0';
        DOCUMENTATION_OPTIONS.show_version_warning_banner = true;
        </script>
    <script src="../../../_static/script/dynamic-nav-dropdown.js?v=5e71fbcf"></script>
    <link rel="canonical" href="https://rtguides.it.tufts.edu/bio/lectures/introduction-to-alphafold2/02-run-alpha-fold2-open-ondemand.html" />
    <link rel="icon" href="../../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="Visualize alphafold2 predicted structure with PYMOL" href="03-vizualize-predicted-structure.html" />
    <link rel="prev" title="Run AlphaFold2 on Tufts HPC with Command Line Interface (CLI)" href="01-run-alpha-fold2-cli.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="20250529" />
    <meta name="docbuild:last-update" content="May 29, 2025"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder=""
         aria-label=""
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
  <aside class="bd-header-announcement d-print-none d-none" aria-label="Announcement" data-pst-announcement-url="https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/announcement/announcement.html"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class="col-lg-3 navbar-header-items__start">
    
      <div class="navbar-item">     
<a class="navbar-brand logo" href="../../../index.html">
           
  <img
    src="../../../_static/jumbo.png"
    class="logo__image only-light"
    alt=""
  />
  <img
    src="../../../_static/jumbo.png"
    class="logo__image only-dark pst-js-only"
    alt=""
  />
   
  <p class="title logo__title" id="long_title">
    TTS Research Technology Guides
  </p>
  
  <p class="title logo__title" id="short_title">TTS RT Guides</p>
   
</a></div>
    
  </div>
  
  <div class="col-lg-9 navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
        </div>
      
      
        <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item"><nav class="bd-docs-nav bd-links" aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">
     Bioinformatics 
  </p>
  <div class="bd-toc-item navbar-nav"><ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../tutorials/index.html">Bioinformatics Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../workshops/index.html">Bioinformatics Workshops</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../databases/index.html">Bioinformatics Databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../apps/index.html">Bioinformatics Applications</a></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../index.html">Bioinformatics Lectures and Trainings</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../igdh-1001/index.html">Introduction to Metagenomics</a></li>
<li class="toctree-l2 current active has-children"><a class="reference internal" href="index.html">Cas12a Mutant Structural Predictions With AlphaFold2</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="01-run-alpha-fold2-cli.html">Run AlphaFold2 on Tufts HPC with Command Line Interface (CLI)</a></li>
<li class="toctree-l3 current active"><a class="current reference internal" href="#">Run AlphaFold2 on Tufts HPC with Open OnDemand App</a></li>
<li class="toctree-l3"><a class="reference internal" href="03-vizualize-predicted-structure.html">Visualize alphafold2 predicted structure with PYMOL</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="../introduction-to-git.html">Introduction to Git&amp;GitHub</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../../resources/index.html">Additional Resources</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"> 

<nav aria-label="Breadcrumb">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a
        href="../../../index.html"
        class="nav-link"
        aria-label="Home"
      >
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
     
    <li class="breadcrumb-item">
      <a href="../../index.html" class="nav-link"
        >Bioinformatics</a
      >
    </li>
     
    <li class="breadcrumb-item">
      <a href="../index.html" class="nav-link"
        >Bioinformatics...</a
      >
    </li>
     
    <li class="breadcrumb-item">
      <a href="index.html" class="nav-link"
        >Cas12a Mutant...</a
      >
    </li>
     
    <li class="breadcrumb-item active" aria-current="page">
      Run AlphaFold2 on...
    </li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="run-alphafold2-on-tufts-hpc-with-open-ondemand-app">
<h1>Run AlphaFold2 on Tufts HPC with Open OnDemand App<a class="headerlink" href="#run-alphafold2-on-tufts-hpc-with-open-ondemand-app" title="Link to this heading">#</a></h1>
<p>This tutorial will guide the user through the process of running AlphaFold2 on Tufts High Performance Computing (HPC) system using Open Ondemand app, a graphical interface to the HPC.</p>
<section id="prerequisites">
<h2>Prerequisites<a class="headerlink" href="#prerequisites" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p>Access to Tufts HPC: If you don’t already have an account, you can apply for a HPC account by submitting this <a class="reference external" href="https://tufts.qualtrics.com/jfe/form/SV_5bUmpFT0IXeyEfj">form</a>.</p></li>
</ul>
</section>
<section id="navigate-to-the-cluster-and-prepare-the-input-file">
<h2>Navigate To The Cluster and prepare the input file<a class="headerlink" href="#navigate-to-the-cluster-and-prepare-the-input-file" title="Link to this heading">#</a></h2>
<p>Navigate to: <a class="reference external" href="https://ondemand.pax.tufts.edu/">https://ondemand.pax.tufts.edu/</a></p>
<ul>
<li><p>Log in with your Tufts credentials</p></li>
<li><p>On the top menu bar choose <code class="docutils literal notranslate"><span class="pre">Clusters-&gt;Tufts</span> <span class="pre">HPC</span> <span class="pre">Shell</span> <span class="pre">Access</span></code></p></li>
<li><p>Type <code class="docutils literal notranslate"><span class="pre">pwd</span></code> to check your current directory. You should be at your home directory, which is listed as <code class="docutils literal notranslate"><span class="pre">/cluster/home/your_utln/</span></code>
<img alt="" src="../../../_images/cluster-screenshot-homepath.png" /></p></li>
<li><p>Type <code class="docutils literal notranslate"><span class="pre">ls</span></code> to list the files in your current directory.</p></li>
<li><p>Copy the example folder and data to your home directory by following command</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>cp<span class="w"> </span>-r<span class="w"> </span>/cluster/tufts/bio/tools/training/cas12a_af2_sp24/<span class="w"> </span>./
</pre></div>
</div>
</li>
<li><p>Type <code class="docutils literal notranslate"><span class="pre">ls</span></code> to list the files in your current directory after copying the files, you can now see the <code class="docutils literal notranslate"><span class="pre">cas12a_af2_sp24/</span></code> folder.<br />
<img alt="" src="../../../_images/cluster-screenshot-homedir.png" /></p></li>
</ul>
</section>
<section id="prepare-your-input-data">
<h2>Prepare your input data<a class="headerlink" href="#prepare-your-input-data" title="Link to this heading">#</a></h2>
<p>Input data for AlphaFold2 typically includes the amino acid sequences of the proteins you wish to model.</p>
<p>Your input data should be located here: <code class="docutils literal notranslate"><span class="pre">/cluster/home/your_utln/cas12a_af2_sp24/5XUS_mut2cwf_modified.fasta</span></code><br />
<em>Remember to replace <code class="docutils literal notranslate"><span class="pre">your_utln</span></code> with your own tufts credential</em></p>
</section>
<section id="run-alphafold2">
<h2>Run AlphaFold2<a class="headerlink" href="#run-alphafold2" title="Link to this heading">#</a></h2>
<section id="navigate-to-https-ondemand-pax-tufts-edu">
<h3>Navigate to: <a class="reference external" href="https://ondemand.pax.tufts.edu/">https://ondemand.pax.tufts.edu/</a><a class="headerlink" href="#navigate-to-https-ondemand-pax-tufts-edu" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Log in with your Tufts credentials</p></li>
<li><p>On the top menu bar choose <code class="docutils literal notranslate"><span class="pre">Bioinformatics</span> <span class="pre">Apps-&gt;AlphaFold</span></code><br />
<img alt="" src="../../../_images/ondemand-screenshot-af0.png" /></p></li>
</ul>
</section>
<section id="enter-the-following-parameters-to-run-the-job">
<h3>Enter the following parameters to run the job.<a class="headerlink" href="#enter-the-following-parameters-to-run-the-job" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">Number</span> <span class="pre">of</span> <span class="pre">hours:</span> <span class="pre">24</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Number</span> <span class="pre">of</span> <span class="pre">cores:</span> <span class="pre">8</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Amount</span> <span class="pre">of</span> <span class="pre">memory:</span> <span class="pre">32GB</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Select</span> <span class="pre">preempt</span> <span class="pre">or</span> <span class="pre">normal</span> <span class="pre">gpu</span> <span class="pre">partition:</span> <span class="pre">gpu</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Select</span> <span class="pre">the</span> <span class="pre">GPU</span> <span class="pre">type:</span> <span class="pre">a100</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Software</span> <span class="pre">Version:</span> <span class="pre">2.3.2</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Database:</span> <span class="pre">20231031</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Working</span> <span class="pre">Directory:</span> <span class="pre">/cluster/home/your_utln/cas12a_af2_sp24/</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Output</span> <span class="pre">directory</span> <span class="pre">Name:</span> <span class="pre">/cluster/home/your_utln/cas12a_af2_sp24/</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">fasta_paths:</span> <span class="pre">/cluster/home/your_utln/cas12a_af2_sp24/5XUS_mut2cwf_modified.fasta</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">model_preset:</span> <span class="pre">multimer</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">models_to_relax:</span> <span class="pre">best</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">num_multimer_predictions_per_model:</span> <span class="pre">1</span></code></p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">max_template_date:</span> <span class="pre">2020-01-01</span></code>
Then, hit <code class="docutils literal notranslate"><span class="pre">launch</span></code><br />
<img alt="" src="../../../_images/ondemand-screenshot-af1.png" /></p></li>
</ul>
</section>
<section id="check-output-files">
<h3>Check output files<a class="headerlink" href="#check-output-files" title="Link to this heading">#</a></h3>
<p>To check the output files, go back to the main page of Open Ondemand, on the top menu bar choose <code class="docutils literal notranslate"><span class="pre">Files</span> <span class="pre">-&gt;</span> <span class="pre">Home</span> <span class="pre">Directory</span></code><br />
Click <code class="docutils literal notranslate"><span class="pre">cas12a_af2_sp24</span></code>, then click <code class="docutils literal notranslate"><span class="pre">5XUS_mut2cwf_modified</span></code>, you will see the output. Given that this task requires a few hours to complete, proceed to the <code class="docutils literal notranslate"><span class="pre">5XUS_mut2cwf_modified_prerun</span></code> folder. This location contains the pre-executed results of AlphaFold2 along with the stored output files.<br />
<img alt="" src="../../../_images/ondemand-screenshot-afoutput.png" /></p>
<p>Here’s the explanation of the file categories listed:</p>
<ul class="simple">
<li><p>JSON Files (<em>confidence_model &amp; pae_model</em>): Predictive data files from different models, indicating confidence levels and pairwise alignment errors.</p></li>
<li><p>Pickle File (<em>features.pkl</em>): Serialized file containing features used for model prediction.
Directory (<em>msas</em>): A folder containing multiple sequence alignments related to the analysis.</p></li>
<li><p>CIF &amp; PDB Files (_ranked_, relaxed_model_, unrelaxed_model__): Structural data files in two formats, showing the ranked and modelled protein structures, both relaxed and unrelaxed.</p></li>
<li><p>JSON Files (<em>ranking_debug.json, relax_metrics.json, timings.json</em>): Debugging, metrics, and timing information for the analysis process.</p></li>
<li><p>Pickle Files (_result_model__): Serialized files containing comprehensive results from each model prediction.</p></li>
</ul>
<p><strong><code class="docutils literal notranslate"><span class="pre">ranked_0.pdb</span></code> is your best predicted structure.</strong><br />
This file contains the highest-ranked prediction based on the analysis, indicating it is considered the most accurate or reliable structure prediction from the set of models used. You should be able to download this file with the <code class="docutils literal notranslate"><span class="pre">download</span></code> button.</p>
</section>
</section>
<section id="alphafold2-accuracy-assessment">
<h2>AlphaFold2 Accuracy Assessment<a class="headerlink" href="#alphafold2-accuracy-assessment" title="Link to this heading">#</a></h2>
<blockquote>
<div><p>[!NOTE]<br />
To continue with this section, you will need to know some linux basics. More information can be found <a class="reference external" href="https://it.tufts.edu/file/introduction-basic-linux">here</a>.</p>
</div></blockquote>
<p>We can assess the accuracy of the AlphaFold prediction using:</p>
<ul class="simple">
<li><p><strong>Predicted Local Distance Difference Test (pLDDT)</strong></p></li>
<li><p><strong>Predicted Alignment Error</strong></p></li>
</ul>
<section id="predicted-local-distance-difference-test-plddt">
<h3>Predicted Local Distance Difference Test (pLDDT)<a class="headerlink" href="#predicted-local-distance-difference-test-plddt" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>per-residue confidence metric ranging from 0-100 (100 being the highest confidence)</p></li>
<li><p>Regions below 50 could indicate disordered regions</p></li>
</ul>
<p><img alt="" src="../../../_images/plddt.png" /></p>
</section>
<section id="predicted-alignment-error-pae">
<h3>Predicted Alignment Error (PAE)<a class="headerlink" href="#predicted-alignment-error-pae" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>The Predicted Alignment Error (PAE) gives us an expected distance error based on each residue.</p></li>
<li><p>If we are more confident that the distance between two residues is accurate, then the PAE is lower (darker green). If we are less confident that the distance between two residues is accurate, the PAE is higher (lighter green)</p></li>
</ul>
<p><img alt="" src="../../../_images/pae.png" /></p>
</section>
<section id="navigate-to-the-cluster">
<h3>Navigate To The Cluster<a class="headerlink" href="#navigate-to-the-cluster" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Now that we have an idea of what these metrics mean, let’s try generating these plots for the Cas12a-CWF mutant on the cluster. First navigate to: <a class="reference external" href="https://ondemand.pax.tufts.edu/">https://ondemand.pax.tufts.edu/</a></p></li>
<li><p>Log in with your Tufts credentials</p></li>
<li><p>On the top menu bar choose <code class="docutils literal notranslate"><span class="pre">Clusters-&gt;Tufts</span> <span class="pre">HPC</span> <span class="pre">Shell</span> <span class="pre">Access</span></code></p></li>
</ul>
<p><img alt="" src="../../../_images/shell.png" /></p>
<ul class="simple">
<li><p>You’ll see a welcome message and a bash prompt, for example for user <code class="docutils literal notranslate"><span class="pre">tutln01</span></code>:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[tutln01@login001 ~]$
</pre></div>
</div>
<ul class="simple">
<li><p>This indicates you are logged in to the login node of the cluster. Please <strong>do not</strong> run any program from the login node.</p></li>
</ul>
</section>
<section id="starting-an-interactive-session">
<h3>Starting an Interactive Session<a class="headerlink" href="#starting-an-interactive-session" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>To run our analyses we will need to move from the login node to a compute node. We can do this by entering:</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>srun<span class="w"> </span>-p<span class="w"> </span>batch<span class="w"> </span>--time<span class="o">=</span><span class="m">3</span>:00:00<span class="w"> </span>-n<span class="w"> </span><span class="m">2</span><span class="w"> </span>--mem<span class="o">=</span>4g<span class="w"> </span>--pty<span class="w"> </span>bash
</pre></div>
</div>
<p><em>Explanation of Commands</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>- `srun`: SLURM command to run a parallel job
- `-p`: asking for a partition, here we are requesting the batch partition
- `--time`: time we need here we request 3 hours
- `-n`:  number of CPUs needed here we requested 2
- `--mem`:  memory we need here we request 4 Gigabytes
- `--reservation`: the reservation of compute resources to use here we use the `chbe165` reservation
- `--pty`: get a pseudo bash terminal
</pre></div>
</div>
<ul class="simple">
<li><p>When you get a compute node you’ll note that your prompt will no longer say login and instead say the name of the node:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[tutln01@c1cmp048 ~]$
</pre></div>
</div>
</section>
<section id="set-up-for-analysis">
<h3>Set Up For Analysis<a class="headerlink" href="#set-up-for-analysis" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Navigate to the folder that we just ran alphafold with:<br />
<em>Remember to replace <code class="docutils literal notranslate"><span class="pre">your_utln</span></code> with your own utln!!!</em></p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span><span class="w"> </span>/cluster/home/your_utln/cas12a_af2_sp24
</pre></div>
</div>
<ul class="simple">
<li><p>Given that AlphaFold2 can take anywhere from a few hours to a few days to run - AlphaFold2 predictions have already been generated for the Cas12a-CWF mutants from our study. We will use a script from the <a class="reference external" href="https://elearning.bits.vib.be/courses/alphafold/lessons/alphafold-on-the-hpc/topic/alphafold-outputs/">VIB Bioinformatics Core</a> to visualize the accuracy of AlphaFold2’s predictions. First we will need to load the software needed to run that script:</p></li>
</ul>
<blockquote>
<div><p>[!WARNING]
Do not change the module version. Stick with <code class="docutils literal notranslate"><span class="pre">alphafold/2.1.1</span></code>.</p>
</div></blockquote>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>module<span class="w"> </span>load<span class="w"> </span>alphafold/2.1.1
</pre></div>
</div>
</section>
<section id="run-the-python-script">
<h3>Run the python script<a class="headerlink" href="#run-the-python-script" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>Now we will need to feed our script three arguments:</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">--input_dir</span></code> input directory with model files</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--output_dir</span></code> output directory to put our plots of model information</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--name</span></code> optional prefix to add to our file names</p></li>
</ul>
</li>
</ul>
<blockquote>
<div><p>[!NOTE]
Before you run the script, make sure you are at <code class="docutils literal notranslate"><span class="pre">/cluster/home/your_utln/cas12a_af2_sp24</span></code></p>
</div></blockquote>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>python<span class="w"> </span>script/af2_accuracy_viz.py<span class="w"> </span>--input_dir<span class="w"> </span>./5XUS_mut2cwf_modified_prerun<span class="w"> </span>--output_dir<span class="w"> </span>./<span class="w"> </span>--name<span class="w"> </span>mut2cwf_modified
</pre></div>
</div>
</section>
<section id="check-the-output-files">
<h3>Check the output files<a class="headerlink" href="#check-the-output-files" title="Link to this heading">#</a></h3>
<ul>
<li><p>Running this will generate two img in your current directory:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">mut2cwf_modified_coverage_LDDT.png</span></code> - plots of your msa coverage and pLDDT scores per residue per model</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">mut2cwf_modified_PAE.png</span></code> - plots of your predicted alignment error</p></li>
</ul>
</li>
<li><p>The following are the pLDDT and PAE scores for the Cas12a-CWF mutant:</p>
<p><code class="docutils literal notranslate"><span class="pre">mut2cwf_modified_coverage_LDDT.png</span></code>
<img alt="" src="../../../_images/mut2cwf-modified-coverage-lddt.png" /><br />
The left-side heat map showcases the Multiple Sequence Alignment (MSA), with each sequence aligned against the input sequences. The color scale reflects the identity score, arranging sequences from top (highest identity) to bottom (lowest identity). Uncovered areas appear white, indicative of subsequences in the database that do not fully align. A black line delineates the extent of sequence coverage relative to the total number of sequences aligned.
The right-side plot displays the predicted LDDT per residue position.</p>
<p><code class="docutils literal notranslate"><span class="pre">mut2cwf_modified_PAE.png</span></code>
<img alt="" src="../../../_images/mut2cwf-modified-pae.png" /></p>
<p>The PAE (Pairwise Absolute Error) plot visually represents the absolute error in the relative positioning of residues, measured in Ångströms, through pairwise comparison. Utilizing a color gradient from blue to red, dark blue indicates an error of 0 Å, while larger errors shift towards dark red. Typically, along the heat map’s diagonal, values are anticipated to be near 0 Å, signifying minimal error. An optimal model is characterized by a predominantly dark blue heat map, denoting very low error across all comparisons.</p>
</li>
</ul>
<blockquote>
<div><p>[!NOTE]
If you encounter any issues or this doesn’t work as expected, please feel free to reach out to Shirley Li, <a class="reference external" href="mailto:xue&#46;li37&#37;&#52;&#48;tufts&#46;edu">xue<span>&#46;</span>li37<span>&#64;</span>tufts<span>&#46;</span>edu</a></p>
</div></blockquote>
</section>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="01-run-alpha-fold2-cli.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Run AlphaFold2 on Tufts HPC with Command Line Interface (CLI)</p>
      </div>
    </a>
    <a class="right-next"
       href="03-vizualize-predicted-structure.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Visualize alphafold2 predicted structure with PYMOL</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item"><div
  id="pst-page-navigation-heading-2"
  class="page-toc tocsection onthispage"
>
  <i class="fa-solid fa-list"></i> Run AlphaFold2 on Tufts HPC with Open OnDemand App
</div>
<nav
  class="bd-toc-nav page-toc"
  aria-labelledby="pst-page-navigation-heading-2"
>
  <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#prerequisites">Prerequisites</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#navigate-to-the-cluster-and-prepare-the-input-file">Navigate To The Cluster and prepare the input file</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-your-input-data">Prepare your input data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#run-alphafold2">Run AlphaFold2</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#navigate-to-https-ondemand-pax-tufts-edu">Navigate to: https://ondemand.pax.tufts.edu/</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#enter-the-following-parameters-to-run-the-job">Enter the following parameters to run the job.</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#check-output-files">Check output files</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#alphafold2-accuracy-assessment">AlphaFold2 Accuracy Assessment</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#predicted-local-distance-difference-test-plddt">Predicted Local Distance Difference Test (pLDDT)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#predicted-alignment-error-pae">Predicted Alignment Error (PAE)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#navigate-to-the-cluster">Navigate To The Cluster</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#starting-an-interactive-session">Starting an Interactive Session</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#set-up-for-analysis">Set Up For Analysis</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#run-the-python-script">Run the python script</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#check-the-output-files">Check the output files</a></li>
</ul>
</li>
</ul>
</nav></div>

  <div class="sidebar-secondary-item">

  
  <div class="tocsection editthispage">
    <a href="https://github.com/TuftsRT/guides/edit/main/source/bio/lectures/introduction-to-alphafold2/02-run-alpha-fold2-open-ondemand.md">
      <i class="fa-solid fa-pencil"></i>
      
      
Suggest Edits 
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item">

  <div class="tocsection sourcelink">
    <a href="../../../_sources/bio/lectures/introduction-to-alphafold2/02-run-alpha-fold2-open-ondemand.md.txt">
      <i class="fa-solid fa-file-lines"></i> Show Source
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item"><div class="tocsection report-error">
  <a
    href="https://github.com/TuftsRT/guides/issues/new?title=Error Report: bio/lectures/introduction-to-alphafold2/02-run-alpha-fold2-open-ondemand"
    class="report-error-link"
    ><i class="fa-solid fa-triangle-exclamation"></i> Report Error
  </a>
</div></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">

  <p class="copyright">
    
      © Copyright 2025, Tufts University.
      <br/>
    
  </p>
</div>
      
        <div class="footer-item"><p class="last-updated">
  Last updated on May 29, 2025.
  <br/>
</p></div>
      
    </div>
  
  
    <div class="footer-items__center">
      
        <div class="footer-item"><p class="disclaimer">
  Linked external resources not affiliated with or endorsed by Tufts University.
  <br />
</p></div>
      
        <div class="footer-item"><div class="footer-links">
  <ul>
    
    <li><a href="https://access.tufts.edu/digital-accessibility-policy">Accessibility</a></li>
    
    <li><a href="https://oeo.tufts.edu/policies-procedures/non-discrimination-statement/">Non-Discrimination</a></li>
    
    <li><a href="https://www.tufts.edu/about/privacy">Privacy</a></li>
    
  </ul>
</div></div>
      
    </div>
  
  
    <div class="footer-items__end">
      
        <div class="footer-item"><div class="switcher-with-label">
  <div class="switcher-label">
    <p>Version:</p>
  </div>
  
<div class="version-switcher__container dropdown pst-js-only">
  <button id="pst-version-switcher-button-2"
    type="button"
    class="version-switcher__button btn btn-sm dropdown-toggle"
    data-bs-toggle="dropdown"
    aria-haspopup="listbox"
    aria-controls="pst-version-switcher-list-2"
    aria-label="Version switcher list"
  >
    Choose version  <!-- this text may get changed later by javascript -->
    <span class="caret"></span>
  </button>
  <div id="pst-version-switcher-list-2"
    class="version-switcher__menu dropdown-menu list-group-flush py-0"
    role="listbox" aria-labelledby="pst-version-switcher-button-2">
    <!-- dropdown will be populated by javascript on page load -->
  </div>
</div>
</div></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>