
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Set Up Conda Environment and Create Jupyter Kernel for scRNA-seq Analysis &#8212; TTS Research Technology Guides</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=8f2a1f02" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/graphviz.css?v=4ae1632d" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/bugfix.css?v=e4cc360a" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/footer-links.css?v=6edf756e" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/gallery.css?v=c2bd73dd" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/mobile.css?v=6b07021c" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/navbar.css?v=3ee3464b" />
    <link rel="stylesheet" type="text/css" href="../../_static/css/switcher.css?v=3e40d84b" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../_static/documentation_options.js?v=8313637b"></script>
    <script src="../../_static/doctools.js?v=9bcbadda"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script async="async" src="https://www.googletagmanager.com/gtag/js?id=G-ZJ5LXFRJ2G"></script>
    <script>
                window.dataLayer = window.dataLayer || [];
                function gtag(){ dataLayer.push(arguments); }
                gtag('js', new Date());
                gtag('config', 'G-ZJ5LXFRJ2G');
            </script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'bio/tutorials/set-up-conda-environment';</script>
    <script>
        DOCUMENTATION_OPTIONS.theme_version = '0.16.0';
        DOCUMENTATION_OPTIONS.theme_switcher_json_url = 'https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/switcher/switcher.json';
        DOCUMENTATION_OPTIONS.theme_switcher_version_match = '1.0.0-dev';
        DOCUMENTATION_OPTIONS.show_version_warning_banner = true;
        </script>
    <link rel="canonical" href="https://rtguides.it.tufts.edu/bio/tutorials/set-up-conda-environment.html" />
    <link rel="icon" href="../../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Using containers with Singularity to ensure reproducible research" href="using-singularity.html" />
    <link rel="prev" title="scRNA-seq analysis with RStudio" href="sc-rna-seq-analysis-with-r.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="20250527" />
    <meta name="docbuild:last-update" content="May 27, 2025"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder=""
         aria-label=""
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
  <aside class="bd-header-announcement d-print-none d-none" aria-label="Announcement" data-pst-announcement-url="https://raw.githubusercontent.com/TuftsRT/guides/refs/heads/announcement/announcement.html"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
<div class="bd-header__inner bd-page-width">
  <button class="pst-navbar-icon sidebar-toggle primary-toggle" aria-label="Site navigation">
    <span class="fa-solid fa-bars"></span>
  </button>
  
  
  <div class="col-lg-3 navbar-header-items__start">
    
      <div class="navbar-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/jumbo.png" class="logo__image only-light" alt=""/>
    <img src="../../_static/jumbo.png" class="logo__image only-dark pst-js-only" alt=""/>
  
  
    <p class="title logo__title">TTS Research Technology Guides</p>
  
</a></div>
    
  </div>
  
  <div class="col-lg-9 navbar-header-items">
    
    <div class="me-auto navbar-header-items__center">
      
        <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
      
    </div>
    
    
    <div class="navbar-header-items__end">
      
        <div class="navbar-item navbar-persistent--container">
          

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
        </div>
      
      
        <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
      
        <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/dev/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
      
    </div>
    
  </div>
  
  
    <div class="navbar-persistent--mobile">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button>
    </div>
  

  
    <button class="pst-navbar-icon sidebar-toggle secondary-toggle" aria-label="On this page">
      <span class="fa-solid fa-outdent"></span>
    </button>
  
</div>

    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
        
          
          
            <div class="navbar-item">
<nav>
  <ul class="bd-navbar-elements navbar-nav">
    
<li class="nav-item ">
  <a class="nav-link nav-internal" href="../../hpc/index.html">
    High-Performance Computing
  </a>
</li>


<li class="nav-item current active">
  <a class="nav-link nav-internal" href="../index.html">
    Bioinformatics
  </a>
</li>


<li class="nav-item ">
  <a class="nav-link nav-external" href="https://go.tufts.edu/geospatial">
    Geospatial
  </a>
</li>

  </ul>
</nav></div>
          
        
      </div>
    
    
    
      <div class="sidebar-header-items__end">
        
          <div class="navbar-item">

<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button></div>
        
          <div class="navbar-item"><ul class="navbar-icon-links"
    aria-label="Icon Links">
        <li class="nav-item">
            
          
          
          
          
          
          
          
          
          
          <a href="https://rtguides.it.tufts.edu/dev/tags/index.html" title="Tags" class="nav-link pst-navbar-icon" rel="noopener" target="_self" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-tags fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Tags</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="https://github.com/TuftsRT/guides" title="GitHub" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-brands fa-github fa-lg" aria-hidden="true"></i>
            <span class="sr-only">GitHub</span></a>
        </li>
        <li class="nav-item">
          
          
          
          
          
          
          
          
          <a href="mailto:tts-research@tufts.edu" title="Email" class="nav-link pst-navbar-icon" rel="noopener" target="_blank" data-bs-toggle="tooltip" data-bs-placement="bottom"><i class="fa-solid fa-envelope fa-lg" aria-hidden="true"></i>
            <span class="sr-only">Email</span></a>
        </li>
</ul></div>
        
      </div>
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item"><nav class="bd-docs-nav bd-links" aria-label="Section Navigation">
  <p class="bd-links__title" role="heading" aria-level="1">
     Bioinformatics 
  </p>
  <div class="bd-toc-item navbar-nav"><ul class="current nav bd-sidenav">
<li class="toctree-l1 current active has-children"><a class="reference internal" href="index.html">Bioinformatics Tutorials</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="bioinformatics-tools-guide.html">Getting started with bioinformatics tools</a></li>
<li class="toctree-l2"><a class="reference internal" href="container-mod.html">Package installation and module generation with container-mod</a></li>
<li class="toctree-l2"><a class="reference internal" href="de-seq2-in-r-ood.html">A Beginner’s Guide to Differential Expression Analysis with DESeq2 in R on Open OnDemand</a></li>
<li class="toctree-l2"><a class="reference internal" href="exploring-python-bioinformatics.html">Exploring Python Bioinformatics Packages with Jupyter Notebook</a></li>
<li class="toctree-l2"><a class="reference internal" href="parallel-computing-in-r.html">Parallel Computing in R for Bioinformatics</a></li>
<li class="toctree-l2"><a class="reference internal" href="parallel-read-alignment.html">Parallel read alignment</a></li>
<li class="toctree-l2"><a class="reference internal" href="running-nfcore-pipelines.html">Running nfcore pipelines on Tufts HPC</a></li>
<li class="toctree-l2"><a class="reference internal" href="sc-rna-seq-analysis-with-r.html">scRNA-seq analysis with RStudio</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Set Up Conda Environment and Create Jupyter Kernel for scRNA-seq Analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="using-singularity.html">Using containers with Singularity to ensure reproducible research</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../workshops/index.html">Bioinformatics Workshops</a></li>
<li class="toctree-l1"><a class="reference internal" href="../databases/index.html">Bioinformatics Databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../apps/index.html">Bioinformatics Applications</a></li>
<li class="toctree-l1"><a class="reference internal" href="../lectures/index.html">Bioinformatics Lectures and Trainings</a></li>
<li class="toctree-l1"><a class="reference internal" href="../resources/index.html">Additional Resources</a></li>
</ul>
</div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        
          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"> 

<nav aria-label="Breadcrumb">
  <ul class="bd-breadcrumbs">
    
    <li class="breadcrumb-item breadcrumb-home">
      <a
        href="../../index.html"
        class="nav-link"
        aria-label="Home"
      >
        <i class="fa-solid fa-home"></i>
      </a>
    </li>
     
    <li class="breadcrumb-item">
      <a href="../index.html" class="nav-link"
        >Bioinformatics</a
      >
    </li>
     
    <li class="breadcrumb-item">
      <a href="index.html" class="nav-link"
        >Bioinformatics Tutorials</a
      >
    </li>
     
    <li class="breadcrumb-item active" aria-current="page">
      Set Up Conda...
    </li>
  </ul>
</nav>
</div>
      
    </div>
  
  
</div>
</div>
              
              
              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="set-up-conda-environment-and-create-jupyter-kernel-for-scrna-seq-analysis">
<h1>Set Up Conda Environment and Create Jupyter Kernel for scRNA-seq Analysis<a class="headerlink" href="#set-up-conda-environment-and-create-jupyter-kernel-for-scrna-seq-analysis" title="Link to this heading">#</a></h1>
<p>Shirley Li, Bioinformatician, TTS Research Technology
<a class="reference external" href="mailto:xue&#46;li37&#37;&#52;&#48;tufts&#46;edu">xue<span>&#46;</span>li37<span>&#64;</span>tufts<span>&#46;</span>edu</a></p>
<p>Date: 2024-11-01</p>
<section id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Link to this heading">#</a></h2>
<p>In this tutorial, you will learn how to:</p>
<ul class="simple">
<li><p>Create a Conda environment for single-cell RNA-seq analysis using Python-only packages.</p></li>
<li><p>Install popular Python packages for scRNA-seq, such as Scanpy, and Scrublet.</p></li>
<li><p>Set up a Jupyter kernel that uses the Conda environment for easy access to the tools in a notebook interface.</p></li>
</ul>
<section id="create-a-conda-environment-for-scrna-seq">
<h3>Create a Conda Environment for scRNA-seq<a class="headerlink" href="#create-a-conda-environment-for-scrna-seq" title="Link to this heading">#</a></h3>
<ol class="arabic simple">
<li><p>Load <code class="docutils literal notranslate"><span class="pre">miniforge</span></code> and <code class="docutils literal notranslate"><span class="pre">conda-env-mod</span></code> module</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">module</span> <span class="n">load</span> <span class="n">miniforge</span><span class="o">/</span><span class="mf">24.7.1</span><span class="o">-</span><span class="n">py312</span>
<span class="n">module</span> <span class="n">load</span> <span class="n">conda</span><span class="o">-</span><span class="n">env</span><span class="o">-</span><span class="n">mod</span><span class="o">/</span><span class="n">default</span>
</pre></div>
</div>
<ol class="arabic" start="2">
<li><p>Configure your conda</p>
<p><strong>Note (steps in this session only needs to be executed ONCE)</strong></p>
<p>Since your home directory has limited storage, it’s recommended to install conda packages in your group research storage space. Follow these steps:</p>
<p>Create two directories in your group research storage space (one for storing the envs, one for storing the pkgs, for example: condaenv, condapkg)</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>mkdir /cluster/tufts/XXXXlab/$USER/condaenv/
mkdir /cluster/tufts/XXXXlab/$USER/condapkg/
</pre></div>
</div>
<p>If you haven’t used conda before on the cluster, create a file named “.condarc” in your home directory.</p>
<p>Now add the following 4 lines to the <code class="docutils literal notranslate"><span class="pre">.condarc</span></code> file in your home directory (modify according to your real path to the directories):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>envs_dirs:
  - /cluster/tufts/XXXXlab/$USER/condaenv/
pkgs_dirs:
  - /cluster/tufts/XXXXlab/$USER/condapkg/
</pre></div>
</div>
<p>After this, your <code class="docutils literal notranslate"><span class="pre">.condarc</span></code> file should look like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>envs_dirs:
  - /cluster/tufts/XXXXlab/$USER/condaenv/
pkgs_dirs:
  - /cluster/tufts/XXXXlab/$USER/condapkg/
channels:
  - bioconda
  - conda-forge
  - defaults
</pre></div>
</div>
</li>
<li><p>Create your conda environment with <code class="docutils literal notranslate"><span class="pre">conda-env-mod</span></code></p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cd /cluster/tufts/XXXXlab/$USER/condaenv/
conda-env-mod create -p scrna_seq_py_env python=3.8  --jupyter
</pre></div>
</div>
<p>​ You will see something like this, and enter <code class="docutils literal notranslate"><span class="pre">y</span></code> to continue</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>  ...

The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-2_gnu
  asttokens          conda-forge/noarch::asttokens-2.4.1-pyhd8ed1ab_0
  bzip2              conda-forge/linux-64::bzip2-1.0.8-hd590300_5
  ca-certificates    conda-forge/linux-64::ca-certificates-2024.7.4-hbcca054_0
  ...

Proceed ([y]/n)? y
</pre></div>
</div>
<p>​ When it’s complete, you will see something like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">...</span>
<span class="n">Preparing</span> <span class="n">transaction</span><span class="p">:</span> <span class="o">...</span><span class="n">working</span><span class="o">...</span> <span class="n">done</span>
<span class="n">Verifying</span> <span class="n">transaction</span><span class="p">:</span> <span class="o">...</span><span class="n">working</span><span class="o">...</span> <span class="n">done</span>
<span class="n">Executing</span> <span class="n">transaction</span><span class="p">:</span> <span class="o">...</span><span class="n">working</span><span class="o">...</span> <span class="n">done</span>
<span class="o">+---------------------------------------------------------------+</span>
<span class="o">|</span> <span class="n">To</span> <span class="n">use</span> <span class="n">this</span> <span class="n">environment</span><span class="p">,</span> <span class="n">load</span> <span class="n">the</span> <span class="n">following</span> <span class="n">modules</span><span class="p">:</span>          <span class="o">|</span>
<span class="o">|</span>     <span class="n">module</span> <span class="n">load</span> <span class="n">use</span><span class="o">.</span><span class="n">own</span>                                       <span class="o">|</span>
<span class="o">|</span>     <span class="n">module</span> <span class="n">load</span> <span class="n">conda</span><span class="o">-</span><span class="n">env</span><span class="o">/</span><span class="n">scrna_seq_py_env</span><span class="o">-</span><span class="n">py3</span><span class="mf">.12.5</span>           <span class="o">|</span>
<span class="o">|</span> <span class="p">(</span><span class="n">then</span> <span class="n">standard</span> <span class="s1">&#39;conda install&#39;</span> <span class="o">/</span> <span class="s1">&#39;pip install&#39;</span> <span class="o">/</span> <span class="n">run</span> <span class="n">scripts</span><span class="p">)</span> <span class="o">|</span>
<span class="o">+---------------------------------------------------------------+</span>
</pre></div>
</div>
</section>
<section id="install-selected-python-packages">
<h3>Install Selected Python Packages<a class="headerlink" href="#install-selected-python-packages" title="Link to this heading">#</a></h3>
<ol class="arabic">
<li><p>Activate conda environment and install new packages</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">module</span> <span class="n">load</span> <span class="n">use</span><span class="o">.</span><span class="n">own</span>
<span class="n">module</span> <span class="n">load</span> <span class="n">conda</span><span class="o">-</span><span class="n">env</span><span class="o">/</span><span class="n">scrna_seq_py_env</span><span class="o">-</span><span class="n">py3</span><span class="mf">.12.5</span>

<span class="n">conda</span> <span class="nb">list</span> <span class="c1"># check packages installed in this environment</span>

<span class="n">pip</span> <span class="n">install</span> <span class="n">jupyter</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">numpy</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">pandas</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">anndata</span>
<span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="n">scanpy</span>
<span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="n">scrublet</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">harmony</span><span class="o">-</span><span class="n">pytorch</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">gseapy</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">scanorama</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">pyscenic</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">scvi</span><span class="o">-</span><span class="n">tools</span>
<span class="n">pip</span> <span class="n">install</span> <span class="o">-</span><span class="n">i</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">test</span><span class="o">.</span><span class="n">pypi</span><span class="o">.</span><span class="n">org</span><span class="o">/</span><span class="n">simple</span><span class="o">/</span> <span class="n">memento</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">pooch</span>
<span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="n">python</span><span class="o">-</span><span class="n">igraph</span>

<span class="n">conda</span> <span class="nb">list</span> <span class="c1"># check again</span>
</pre></div>
</div>
</li>
<li><p>Create a jupyter kernel</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span><span class="o">-</span><span class="n">env</span><span class="o">-</span><span class="n">mod</span> <span class="n">kernel</span> <span class="o">-</span><span class="n">n</span> <span class="n">scrna_seq_py_env</span>
</pre></div>
</div>
<p>You will see something like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">requested</span> <span class="n">kernel</span> <span class="k">with</span> <span class="n">arguments</span><span class="p">:</span>  <span class="o">-</span><span class="n">n</span> <span class="s1">&#39;scrna_seq_py_env&#39;</span> <span class="o">--</span>

<span class="n">Jupyter</span> <span class="n">kernel</span> <span class="n">created</span><span class="p">:</span> <span class="s2">&quot;Python (My scrna_seq_py_env Kernel)&quot;</span>
<span class="o">+---------------------------------------------------------------+</span>
<span class="o">|</span> <span class="n">We</span> <span class="n">recommend</span> <span class="n">installing</span> <span class="n">packages</span> <span class="n">into</span> <span class="n">your</span> <span class="n">kernel</span> <span class="n">environment</span> <span class="o">|</span>
<span class="o">|</span> <span class="n">via</span> <span class="n">the</span> <span class="n">command</span> <span class="n">line</span> <span class="p">(</span><span class="k">with</span> <span class="s1">&#39;conda install&#39;</span> <span class="ow">or</span> <span class="s1">&#39;pip install&#39;</span><span class="p">)</span><span class="o">.</span> <span class="o">|</span>
<span class="o">+---------------------------------------------------------------+</span>
</pre></div>
</div>
</li>
</ol>
</section>
</section>
<section id="using-open-ondemand-jupyter-lab">
<h2>Using Open OnDemand Jupyter Lab<a class="headerlink" href="#using-open-ondemand-jupyter-lab" title="Link to this heading">#</a></h2>
<p>Natigate to <a class="reference external" href="https://ondemand.pax.tufts.edu/">Open Ondemand</a></p>
<p>In Open Ondemand dashboard, let’s go to <code class="docutils literal notranslate"><span class="pre">Interactive</span> <span class="pre">APPs</span></code> =&gt; <code class="docutils literal notranslate"><span class="pre">Jupyter</span> <span class="pre">Lab</span></code> and select the <code class="docutils literal notranslate"><span class="pre">number</span> <span class="pre">of</span> <span class="pre">hours</span></code>, <code class="docutils literal notranslate"><span class="pre">number</span> <span class="pre">of</span> <span class="pre">cores</span></code>, and <code class="docutils literal notranslate"><span class="pre">Amount</span> <span class="pre">of</span> <span class="pre">memory</span></code> that you would like to request and Launch this job.</p>
<p>Under <code class="docutils literal notranslate"><span class="pre">Notebook</span></code>, select the kernel you just created. Ex: <code class="docutils literal notranslate"><span class="pre">scrna_seq_py_env</span></code></p>
<p>Start your python code from there.</p>
<p>Example code to check the installation:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Import installed packages</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">os</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">seaborn</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">sns</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">scanpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">sc</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">scrublet</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">scr</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">anndata</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">harmony</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">memento</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">scvi</span>
<span class="kn">import</span><span class="w"> </span><span class="nn">matplotlib.pyplot</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">plt</span>
</pre></div>
</div>
</section>
<section id="single-cell-rna-seq-analysis-packages">
<h2>Single-Cell RNA-seq Analysis Packages<a class="headerlink" href="#single-cell-rna-seq-analysis-packages" title="Link to this heading">#</a></h2>
<section id="scanpy">
<h3>Scanpy<a class="headerlink" href="#scanpy" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Summary</strong>: <code class="docutils literal notranslate"><span class="pre">Scanpy</span></code> is a widely used Python package for analyzing large-scale single-cell RNA-seq datasets. It is optimized for scalability and supports workflows for preprocessing, clustering, dimensionality reduction, differential expression, and visualization of single-cell data.</p></li>
<li><p><strong>Paper</strong>: Wolf, F. A., Angerer, P., &amp; Theis, F. J. (2018). “Scanpy: large-scale single-cell gene expression data analysis.” <em>Genome Biology</em>, 19(1), 15. <a class="reference external" href="https://doi.org/10.1186/s13059-017-1382-0">https://doi.org/10.1186/s13059-017-1382-0</a> % codespell:ignore theis</p></li>
<li><p><strong>Website</strong>: <a class="reference external" href="https://scanpy.readthedocs.io">https://scanpy.readthedocs.io</a></p></li>
</ul>
</section>
<section id="scrublet">
<h3>Scrublet<a class="headerlink" href="#scrublet" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Summary</strong>: <code class="docutils literal notranslate"><span class="pre">Scrublet</span></code> is a Python tool designed to detect doublets in single-cell RNA-seq data. Doublets are instances where two cells are captured in a single droplet, which can distort downstream analysis. Scrublet uses a k-nearest neighbors approach to identify and score potential doublets.</p></li>
<li><p><strong>Paper</strong>: Wolock, S. L., Lopez, R., &amp; Klein, A. M. (2019). “Scrublet: Computational Identification of Cell Doublets in Single-Cell Transcriptomic Data.” <em>Cell Systems</em>, 8(4), 281–291.e9. <a class="reference external" href="https://doi.org/10.1016/j.cels.2018.11.005">https://doi.org/10.1016/j.cels.2018.11.005</a></p></li>
<li><p><strong>Website</strong>: <a class="github reference external" href="https://github.com/AllonKleinLab/scrublet">AllonKleinLab/scrublet</a></p></li>
</ul>
</section>
<section id="anndata">
<h3>AnnData<a class="headerlink" href="#anndata" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Summary</strong>: <code class="docutils literal notranslate"><span class="pre">AnnData</span></code> is a Python package that provides a framework for managing annotated data matrices, tailored for large-scale single-cell RNA-seq data. AnnData is widely used as the primary data structure in <code class="docutils literal notranslate"><span class="pre">Scanpy</span></code>, enabling efficient storage and handling of both raw and processed single-cell data.</p></li>
<li><p><strong>Paper</strong>: Virshup, I., Rybakov, S., Theis, F. J., Angerer, P., &amp; Wolf, F. A. (2024). “anndata: Access and store annotated data matrices.” <em>The Journal of Open Source Software</em>. <a class="reference external" href="https://doi.org/10.21105/joss.04371">https://doi.org/10.21105/joss.04371</a> % codespell:ignore theis</p></li>
<li><p><strong>Website</strong>: <a class="reference external" href="https://anndata.readthedocs.io">https://anndata.readthedocs.io</a></p></li>
</ul>
</section>
<section id="harmony">
<h3>Harmony<a class="headerlink" href="#harmony" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Summary</strong>: <code class="docutils literal notranslate"><span class="pre">Harmony</span></code> is a tool designed for batch effect correction in single-cell RNA-seq datasets. It integrates datasets from different batches or conditions by aligning data in a shared embedding space, allowing biological variation to be preserved while minimizing technical differences.</p></li>
<li><p><strong>Paper</strong>: Korsunsky, I., Millard, N., Fan, J., Slowikowski, K., Zhang, F., Wei, K., … &amp; Raychaudhuri, S. (2019). “Fast, sensitive and accurate integration of single-cell data with Harmony.” <em>Nature Methods</em>, 16(12), 1289-1296. <a class="reference external" href="https://doi.org/10.1038/s41592-019-0619-0">https://doi.org/10.1038/s41592-019-0619-0</a></p></li>
<li><p><strong>Website</strong>: <a class="reference external" href="https://portals.broadinstitute.org/harmony">https://portals.broadinstitute.org/harmony</a></p></li>
</ul>
</section>
<section id="memento">
<h3>Memento<a class="headerlink" href="#memento" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Summary</strong>: <code class="docutils literal notranslate"><span class="pre">Memento</span></code> is a statistical tool tailored for single-cell RNA sequencing (scRNA-seq) analysis, with a focus on decoupling measurement noise from biological expression variability, thereby improving accuracy in differential expression studies.</p></li>
<li><p><strong>Paper</strong>: Kim, M. C., Gate, R., Lee, D. S., Marson, A., Ntranos, V., Ye, C. J. (2024). “Method of moments framework for differential expression analysis of single-cell RNA sequencing data.” <em>Cell</em>, 187(22), P6393-6410.E16. <a class="reference external" href="https://doi.org/10.1016/j.cell.2024.08.022">https://doi.org/10.1016/j.cell.2024.08.022</a><a class="reference external" href="https://doi.org/10.1038/s41592-021-01125-y">https://doi.org/10.1038/s41592-021-01125-y</a>)</p></li>
<li><p><strong>Website</strong>: <a class="github reference external" href="https://github.com/yelabucsf/scrna-parameter-estimation">yelabucsf/scrna-parameter-estimation</a></p></li>
</ul>
</section>
<section id="scvi-tools">
<h3>scVI-tools<a class="headerlink" href="#scvi-tools" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Summary</strong>: <code class="docutils literal notranslate"><span class="pre">scVI-tools</span></code> is a framework built on top of PyTorch for scalable probabilistic modeling of single-cell data. It includes various models like scVI (single-cell variational inference), totalVI, and PEAKVI, used for data integration, dimensionality reduction, differential expression, and multi-omics data analysis.</p></li>
<li><p><strong>Paper</strong>: Gayoso, A., Lopez, R., Xing, G., Boyeau, P., Wu, K., Jayasuriya, M., et al. (2022). “A Python library for probabilistic analysis of single-cell omics data.” <em>Nature Biotechnology</em>, 40, 163–166. <a class="reference external" href="https://www.nature.com/articles/s41587-021-01206-w">https://www.nature.com/articles/s41587-021-01206-w</a></p></li>
<li><p><strong>Website</strong>: <a class="reference external" href="https://scvi-tools.org">https://scvi-tools.org</a></p></li>
</ul>
</section>
</section>
</section>


                </article>
              
              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="sc-rna-seq-analysis-with-r.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">scRNA-seq analysis with RStudio</p>
      </div>
    </a>
    <a class="right-next"
       href="using-singularity.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Using containers with Singularity to ensure reproducible research</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item"><div
  id="pst-page-navigation-heading-2"
  class="page-toc tocsection onthispage"
>
  <i class="fa-solid fa-list"></i> Set Up Conda Environment and Create Jupyter Kernel for scRNA-seq Analysis
</div>
<nav
  class="bd-toc-nav page-toc"
  aria-labelledby="pst-page-navigation-heading-2"
>
  <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#overview">Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#create-a-conda-environment-for-scrna-seq">Create a Conda Environment for scRNA-seq</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#install-selected-python-packages">Install Selected Python Packages</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#using-open-ondemand-jupyter-lab">Using Open OnDemand Jupyter Lab</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#single-cell-rna-seq-analysis-packages">Single-Cell RNA-seq Analysis Packages</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#scanpy">Scanpy</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#scrublet">Scrublet</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#anndata">AnnData</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#harmony">Harmony</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#memento">Memento</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#scvi-tools">scVI-tools</a></li>
</ul>
</li>
</ul>
</nav></div>

  <div class="sidebar-secondary-item"> 
<div class="tocsection tags-label">
  <i class="fa-solid fa-tags"></i> Tags
</div>
<ul class="tocsection tags-list">
  <li class="tag-item">
    <a href="../../tags/bioinformatics.html" class="tag-link">bioinformatics</a>
  </li>
</ul>
 </div>

  <div class="sidebar-secondary-item">

  
  <div class="tocsection editthispage">
    <a href="https://github.com/TuftsRT/guides/edit/develop/source/bio/tutorials/set-up-conda-environment.md">
      <i class="fa-solid fa-pencil"></i>
      
      
Suggest Edits 
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item">

  <div class="tocsection sourcelink">
    <a href="../../_sources/bio/tutorials/set-up-conda-environment.md.txt">
      <i class="fa-solid fa-file-lines"></i> Show Source
    </a>
  </div>
</div>

  <div class="sidebar-secondary-item"><div class="tocsection report-error">
  <a
    href="https://github.com/TuftsRT/guides/issues/new?title=Error Report: bio/tutorials/set-up-conda-environment"
    class="report-error-link"
    ><i class="fa-solid fa-triangle-exclamation"></i> Report Error
  </a>
</div></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
          </footer>
        
      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
<div class="bd-footer__inner bd-page-width">
  
    <div class="footer-items__start">
      
        <div class="footer-item">

  <p class="copyright">
    
      © Copyright 2025, Tufts University.
      <br/>
    
  </p>
</div>
      
        <div class="footer-item"><p class="last-updated">
  Last updated on May 27, 2025.
  <br/>
</p></div>
      
    </div>
  
  
    <div class="footer-items__center">
      
        <div class="footer-item"><p class="disclaimer">
  Linked external resources not affiliated with or endorsed by Tufts University.
  <br />
</p></div>
      
        <div class="footer-item"><div class="footer-links">
  <ul>
    
    <li><a href="https://access.tufts.edu/digital-accessibility-policy">Accessibility</a></li>
    
    <li><a href="https://oeo.tufts.edu/policies-procedures/non-discrimination-statement/">Non-Discrimination</a></li>
    
    <li><a href="https://www.tufts.edu/about/privacy">Privacy</a></li>
    
  </ul>
</div></div>
      
    </div>
  
  
    <div class="footer-items__end">
      
        <div class="footer-item"><div class="switcher-with-label">
  <div class="switcher-label">
    <p>Version:</p>
  </div>
  
<div class="version-switcher__container dropdown pst-js-only">
  <button id="pst-version-switcher-button-2"
    type="button"
    class="version-switcher__button btn btn-sm dropdown-toggle"
    data-bs-toggle="dropdown"
    aria-haspopup="listbox"
    aria-controls="pst-version-switcher-list-2"
    aria-label="Version switcher list"
  >
    Choose version  <!-- this text may get changed later by javascript -->
    <span class="caret"></span>
  </button>
  <div id="pst-version-switcher-list-2"
    class="version-switcher__menu dropdown-menu list-group-flush py-0"
    role="listbox" aria-labelledby="pst-version-switcher-button-2">
    <!-- dropdown will be populated by javascript on page load -->
  </div>
</div>
</div></div>
      
    </div>
  
</div>

  </footer>
  </body>
</html>